{
  "cells": [
    {
      "cell_type": "markdown",
      "metadata": {
        "id": "TWZgol8BGeIU"
      },
      "source": [
        "# Etapa 0 - Instalando Bibliotecas\n",
        "Instalando as bibliotecas que serão utilizadas para o funcionamento do código, junto do download dos arquivos que serão utilizados no pipeline"
      ]
    },
    {
      "cell_type": "code",
      "source": [
        "# Método mais fácil de obter os arquivos a partir do drive\n",
        "# Chr12-reference.fasta.gz\n",
        "# caso2-sindrome-metabolica_R1.fastq.gz\n",
        "# caso2-sindrome-metabolica_R2.fastq.gz\n",
        "!pip install gdown\n",
        "!gdown --id 187SiNPJbvcesW6WwgFhY3GVvKTRALsmO\n",
        "!gdown --id 11MfHgoj2UhQEaAlppIjvGETQU-V0Xf_P\n",
        "!gdown --id 1sDYvGKhZxkBcmshK6L1Ph92U3Kcv1i2W"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "au_qgmipL60m",
        "outputId": "9da18072-b9c2-413e-fa6f-5ec71731bde1",
        "collapsed": true
      },
      "execution_count": null,
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "Requirement already satisfied: gdown in /usr/local/lib/python3.12/dist-packages (5.2.0)\n",
            "Requirement already satisfied: beautifulsoup4 in /usr/local/lib/python3.12/dist-packages (from gdown) (4.13.5)\n",
            "Requirement already satisfied: filelock in /usr/local/lib/python3.12/dist-packages (from gdown) (3.20.0)\n",
            "Requirement already satisfied: requests[socks] in /usr/local/lib/python3.12/dist-packages (from gdown) (2.32.4)\n",
            "Requirement already satisfied: tqdm in /usr/local/lib/python3.12/dist-packages (from gdown) (4.67.1)\n",
            "Requirement already satisfied: soupsieve>1.2 in /usr/local/lib/python3.12/dist-packages (from beautifulsoup4->gdown) (2.8)\n",
            "Requirement already satisfied: typing-extensions>=4.0.0 in /usr/local/lib/python3.12/dist-packages (from beautifulsoup4->gdown) (4.15.0)\n",
            "Requirement already satisfied: charset_normalizer<4,>=2 in /usr/local/lib/python3.12/dist-packages (from requests[socks]->gdown) (3.4.4)\n",
            "Requirement already satisfied: idna<4,>=2.5 in /usr/local/lib/python3.12/dist-packages (from requests[socks]->gdown) (3.11)\n",
            "Requirement already satisfied: urllib3<3,>=1.21.1 in /usr/local/lib/python3.12/dist-packages (from requests[socks]->gdown) (2.5.0)\n",
            "Requirement already satisfied: certifi>=2017.4.17 in /usr/local/lib/python3.12/dist-packages (from requests[socks]->gdown) (2025.10.5)\n",
            "Requirement already satisfied: PySocks!=1.5.7,>=1.5.6 in /usr/local/lib/python3.12/dist-packages (from requests[socks]->gdown) (1.7.1)\n",
            "/usr/local/lib/python3.12/dist-packages/gdown/__main__.py:140: FutureWarning: Option `--id` was deprecated in version 4.3.1 and will be removed in 5.0. You don't need to pass it anymore to use a file ID.\n",
            "  warnings.warn(\n",
            "Downloading...\n",
            "From (original): https://drive.google.com/uc?id=187SiNPJbvcesW6WwgFhY3GVvKTRALsmO\n",
            "From (redirected): https://drive.google.com/uc?id=187SiNPJbvcesW6WwgFhY3GVvKTRALsmO&confirm=t&uuid=87e92f27-8b79-4b86-9fa9-d41ee2f1cfb9\n",
            "To: /content/caso2-sindrome-metabolica_R1.fastq.gz\n",
            "100% 424M/424M [00:06<00:00, 65.1MB/s]\n",
            "/usr/local/lib/python3.12/dist-packages/gdown/__main__.py:140: FutureWarning: Option `--id` was deprecated in version 4.3.1 and will be removed in 5.0. You don't need to pass it anymore to use a file ID.\n",
            "  warnings.warn(\n",
            "Downloading...\n",
            "From (original): https://drive.google.com/uc?id=11MfHgoj2UhQEaAlppIjvGETQU-V0Xf_P\n",
            "From (redirected): https://drive.google.com/uc?id=11MfHgoj2UhQEaAlppIjvGETQU-V0Xf_P&confirm=t&uuid=7be4a112-9c69-4e96-ba5d-8e995e00587f\n",
            "To: /content/caso2-sindrome-metabolica_R2.fastq.gz\n",
            "100% 424M/424M [00:04<00:00, 88.1MB/s]\n",
            "/usr/local/lib/python3.12/dist-packages/gdown/__main__.py:140: FutureWarning: Option `--id` was deprecated in version 4.3.1 and will be removed in 5.0. You don't need to pass it anymore to use a file ID.\n",
            "  warnings.warn(\n",
            "Downloading...\n",
            "From (original): https://drive.google.com/uc?id=1sDYvGKhZxkBcmshK6L1Ph92U3Kcv1i2W\n",
            "From (redirected): https://drive.google.com/uc?id=1sDYvGKhZxkBcmshK6L1Ph92U3Kcv1i2W&confirm=t&uuid=f6cea69a-ac22-49a6-84ed-7d402a55af60\n",
            "To: /content/Chr12-reference.fasta.gz\n",
            "100% 39.5M/39.5M [00:00<00:00, 63.3MB/s]\n"
          ]
        }
      ]
    },
    {
      "cell_type": "code",
      "execution_count": null,
      "metadata": {
        "collapsed": true,
        "id": "7AAbue9bEe0m",
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "outputId": "0fa4b2b6-75df-4a2d-aea4-6ef8c9432044"
      },
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "PREFIX=/usr/local\n",
            "Unpacking payload ...\n",
            "Extracting _libgcc_mutex-0.1-conda_forge.tar.bz2\n",
            "Extracting ca-certificates-2025.7.14-hbd8a1cb_0.conda\n",
            "Extracting ld_impl_linux-64-2.44-h1423503_1.conda\n",
            "Extracting libgomp-15.1.0-h767d61c_3.conda\n",
            "Extracting pybind11-abi-4-hd8ed1ab_3.tar.bz2\n",
            "Extracting python_abi-3.12-8_cp312.conda\n",
            "Extracting tzdata-2025b-h78e105d_0.conda\n",
            "Extracting _openmp_mutex-4.5-2_gnu.tar.bz2\n",
            "Extracting libgcc-15.1.0-h767d61c_3.conda\n",
            "Extracting c-ares-1.34.5-hb9d3cd8_0.conda\n",
            "Extracting libexpat-2.7.1-hecca717_0.conda\n",
            "Extracting libffi-3.4.6-h2dba641_1.conda\n",
            "Extracting libgcc-ng-15.1.0-h69a702a_3.conda\n",
            "Extracting libiconv-1.18-h4ce23a2_1.conda\n",
            "Extracting liblzma-5.8.1-hb9d3cd8_2.conda\n",
            "Extracting libnsl-2.0.1-hb9d3cd8_1.conda\n",
            "Extracting libstdcxx-15.1.0-h8f9b012_3.conda\n",
            "Extracting libzlib-1.3.1-hb9d3cd8_2.conda\n",
            "Extracting ncurses-6.5-h2d0b736_3.conda\n",
            "Extracting openssl-3.5.1-h7b32b05_0.conda\n",
            "Extracting reproc-14.2.5.post0-hb9d3cd8_0.conda\n",
            "Extracting bzip2-1.0.8-h4bc722e_7.conda\n",
            "Extracting cpp-expected-1.1.0-hff21bea_1.conda\n",
            "Extracting fmt-11.1.4-h07f6e7f_1.conda\n",
            "Extracting keyutils-1.6.1-h166bdaf_0.tar.bz2\n",
            "Extracting libedit-3.1.20250104-pl5321h7949ede_0.conda\n",
            "Extracting libev-4.33-hd590300_2.conda\n",
            "Extracting libsolv-0.7.34-h9463b59_0.conda\n",
            "Extracting libssh2-1.11.1-hcf80075_0.conda\n",
            "Extracting libstdcxx-ng-15.1.0-h4852527_3.conda\n",
            "Extracting libuuid-2.38.1-h0b41bf4_0.conda\n",
            "Extracting libxcrypt-4.4.36-hd590300_1.conda\n",
            "Extracting lz4-c-1.10.0-h5888daf_1.conda\n",
            "Extracting lzo-2.10-hd590300_1001.conda\n",
            "Extracting readline-8.2-h8c095d6_2.conda\n",
            "Extracting reproc-cpp-14.2.5.post0-h5888daf_0.conda\n",
            "Extracting simdjson-3.12.3-h84d6215_0.conda\n",
            "Extracting tk-8.6.13-noxft_hd72426e_102.conda\n",
            "Extracting yaml-cpp-0.8.0-h3f2d84a_0.conda\n",
            "Extracting zstd-1.5.7-hb8e6e7a_2.conda\n",
            "Extracting icu-75.1-he02047a_0.conda\n",
            "Extracting krb5-1.21.3-h659f571_0.conda\n",
            "Extracting libnghttp2-1.64.0-h161d5f1_0.conda\n",
            "Extracting nlohmann_json-3.11.3-he02047a_1.conda\n",
            "Extracting libcurl-8.14.1-h332b0f4_0.conda\n",
            "Extracting libsqlite-3.50.3-hee844dc_1.conda\n",
            "Extracting libxml2-2.13.8-h4bc477f_0.conda\n",
            "Extracting libarchive-3.7.7-h75ea233_4.conda\n",
            "Extracting python-3.12.11-h9e4cc4f_0_cpython.conda\n",
            "Extracting libmamba-2.1.1-h430c389_0.conda\n",
            "Extracting menuinst-2.3.1-py312h7900ff3_0.conda\n",
            "Extracting archspec-0.2.5-pyhd8ed1ab_0.conda\n",
            "Extracting boltons-25.0.0-pyhd8ed1ab_0.conda\n",
            "Extracting brotli-python-1.1.0-py312h2ec8cdc_3.conda\n",
            "Extracting certifi-2025.7.14-pyhd8ed1ab_0.conda\n",
            "Extracting charset-normalizer-3.4.2-pyhd8ed1ab_0.conda\n",
            "Extracting colorama-0.4.6-pyhd8ed1ab_1.conda\n",
            "Extracting distro-1.9.0-pyhd8ed1ab_1.conda\n",
            "Extracting frozendict-2.4.6-py312h66e93f0_0.conda\n",
            "Extracting hpack-4.1.0-pyhd8ed1ab_0.conda\n",
            "Extracting hyperframe-6.1.0-pyhd8ed1ab_0.conda\n",
            "Extracting idna-3.10-pyhd8ed1ab_1.conda\n",
            "Extracting jsonpointer-3.0.0-py312h7900ff3_1.conda\n",
            "Extracting libmambapy-2.1.1-py312h07448e0_0.conda\n",
            "Extracting mamba-2.1.1-had4a41a_0.conda\n",
            "Extracting packaging-25.0-pyh29332c3_1.conda\n",
            "Extracting platformdirs-4.3.8-pyhe01879c_0.conda\n",
            "Extracting pluggy-1.6.0-pyhd8ed1ab_0.conda\n",
            "Extracting pycosat-0.6.6-py312h66e93f0_2.conda\n",
            "Extracting pycparser-2.22-pyh29332c3_1.conda\n",
            "Extracting pysocks-1.7.1-pyha55dd90_7.conda\n",
            "Extracting ruamel.yaml.clib-0.2.8-py312h66e93f0_1.conda\n",
            "Extracting setuptools-80.9.0-pyhff2d567_0.conda\n",
            "Extracting truststore-0.10.1-pyh29332c3_0.conda\n",
            "Extracting wheel-0.45.1-pyhd8ed1ab_1.conda\n",
            "Extracting cffi-1.17.1-py312h06ac9bb_0.conda\n",
            "Extracting h2-4.2.0-pyhd8ed1ab_0.conda\n",
            "Extracting jsonpatch-1.33-pyhd8ed1ab_1.conda\n",
            "Extracting pip-25.1.1-pyh8b19718_0.conda\n",
            "Extracting ruamel.yaml-0.18.14-py312h66e93f0_0.conda\n",
            "Extracting tqdm-4.67.1-pyhd8ed1ab_1.conda\n",
            "Extracting zstandard-0.23.0-py312h66e93f0_2.conda\n",
            "Extracting conda-package-streaming-0.12.0-pyhd8ed1ab_0.conda\n",
            "Extracting urllib3-2.5.0-pyhd8ed1ab_0.conda\n",
            "Extracting requests-2.32.4-pyhd8ed1ab_0.conda\n",
            "Extracting conda-package-handling-2.4.0-pyh7900ff3_2.conda\n",
            "Extracting conda-25.3.1-py312h7900ff3_1.conda\n",
            "Extracting conda-libmamba-solver-25.3.0-pyhd8ed1ab_0.conda\n",
            "\n",
            "Installing base environment...\n",
            "\n",
            "Transaction\n",
            "\n",
            "  Prefix: /usr/local\n",
            "\n",
            "  Updating specs:\n",
            "\n",
            "   - _libgcc_mutex==0.1=conda_forge\n",
            "   - ca-certificates==2025.7.14=hbd8a1cb_0\n",
            "   - ld_impl_linux-64==2.44=h1423503_1\n",
            "   - libgomp==15.1.0=h767d61c_3\n",
            "   - pybind11-abi==4=hd8ed1ab_3\n",
            "   - python_abi==3.12=8_cp312\n",
            "   - tzdata==2025b=h78e105d_0\n",
            "   - _openmp_mutex==4.5=2_gnu\n",
            "   - libgcc==15.1.0=h767d61c_3\n",
            "   - c-ares==1.34.5=hb9d3cd8_0\n",
            "   - libexpat==2.7.1=hecca717_0\n",
            "   - libffi==3.4.6=h2dba641_1\n",
            "   - libgcc-ng==15.1.0=h69a702a_3\n",
            "   - libiconv==1.18=h4ce23a2_1\n",
            "   - liblzma==5.8.1=hb9d3cd8_2\n",
            "   - libnsl==2.0.1=hb9d3cd8_1\n",
            "   - libstdcxx==15.1.0=h8f9b012_3\n",
            "   - libzlib==1.3.1=hb9d3cd8_2\n",
            "   - ncurses==6.5=h2d0b736_3\n",
            "   - openssl==3.5.1=h7b32b05_0\n",
            "   - reproc==14.2.5.0post0=hb9d3cd8_0\n",
            "   - bzip2==1.0.8=h4bc722e_7\n",
            "   - cpp-expected==1.1.0=hff21bea_1\n",
            "   - fmt==11.1.4=h07f6e7f_1\n",
            "   - keyutils==1.6.1=h166bdaf_0\n",
            "   - libedit==3.1.20250104=pl5321h7949ede_0\n",
            "   - libev==4.33=hd590300_2\n",
            "   - libsolv==0.7.34=h9463b59_0\n",
            "   - libssh2==1.11.1=hcf80075_0\n",
            "   - libstdcxx-ng==15.1.0=h4852527_3\n",
            "   - libuuid==2.38.1=h0b41bf4_0\n",
            "   - libxcrypt==4.4.36=hd590300_1\n",
            "   - lz4-c==1.10.0=h5888daf_1\n",
            "   - lzo==2.10=hd590300_1001\n",
            "   - readline==8.2=h8c095d6_2\n",
            "   - reproc-cpp==14.2.5.0post0=h5888daf_0\n",
            "   - simdjson==3.12.3=h84d6215_0\n",
            "   - tk==8.6.13=noxft_hd72426e_102\n",
            "   - yaml-cpp==0.8.0=h3f2d84a_0\n",
            "   - zstd==1.5.7=hb8e6e7a_2\n",
            "   - icu==75.1=he02047a_0\n",
            "   - krb5==1.21.3=h659f571_0\n",
            "   - libnghttp2==1.64.0=h161d5f1_0\n",
            "   - nlohmann_json==3.11.3=he02047a_1\n",
            "   - libcurl==8.14.1=h332b0f4_0\n",
            "   - libsqlite==3.50.3=hee844dc_1\n",
            "   - libxml2==2.13.8=h4bc477f_0\n",
            "   - libarchive==3.7.7=h75ea233_4\n",
            "   - python==3.12.11=h9e4cc4f_0_cpython\n",
            "   - libmamba==2.1.1=h430c389_0\n",
            "   - menuinst==2.3.1=py312h7900ff3_0\n",
            "   - archspec==0.2.5=pyhd8ed1ab_0\n",
            "   - boltons==25.0.0=pyhd8ed1ab_0\n",
            "   - brotli-python==1.1.0=py312h2ec8cdc_3\n",
            "   - certifi==2025.7.14=pyhd8ed1ab_0\n",
            "   - charset-normalizer==3.4.2=pyhd8ed1ab_0\n",
            "   - colorama==0.4.6=pyhd8ed1ab_1\n",
            "   - distro==1.9.0=pyhd8ed1ab_1\n",
            "   - frozendict==2.4.6=py312h66e93f0_0\n",
            "   - hpack==4.1.0=pyhd8ed1ab_0\n",
            "   - hyperframe==6.1.0=pyhd8ed1ab_0\n",
            "   - idna==3.10=pyhd8ed1ab_1\n",
            "   - jsonpointer==3.0.0=py312h7900ff3_1\n",
            "   - libmambapy==2.1.1=py312h07448e0_0\n",
            "   - mamba==2.1.1=had4a41a_0\n",
            "   - packaging==25.0=pyh29332c3_1\n",
            "   - platformdirs==4.3.8=pyhe01879c_0\n",
            "   - pluggy==1.6.0=pyhd8ed1ab_0\n",
            "   - pycosat==0.6.6=py312h66e93f0_2\n",
            "   - pycparser==2.22=pyh29332c3_1\n",
            "   - pysocks==1.7.1=pyha55dd90_7\n",
            "   - ruamel.yaml.clib==0.2.8=py312h66e93f0_1\n",
            "   - setuptools==80.9.0=pyhff2d567_0\n",
            "   - truststore==0.10.1=pyh29332c3_0\n",
            "   - wheel==0.45.1=pyhd8ed1ab_1\n",
            "   - cffi==1.17.1=py312h06ac9bb_0\n",
            "   - h2==4.2.0=pyhd8ed1ab_0\n",
            "   - jsonpatch==1.33=pyhd8ed1ab_1\n",
            "   - pip==25.1.1=pyh8b19718_0\n",
            "   - ruamel.yaml==0.18.14=py312h66e93f0_0\n",
            "   - tqdm==4.67.1=pyhd8ed1ab_1\n",
            "   - zstandard==0.23.0=py312h66e93f0_2\n",
            "   - conda-package-streaming==0.12.0=pyhd8ed1ab_0\n",
            "   - urllib3==2.5.0=pyhd8ed1ab_0\n",
            "   - requests==2.32.4=pyhd8ed1ab_0\n",
            "   - conda-package-handling==2.4.0=pyh7900ff3_2\n",
            "   - conda==25.3.1=py312h7900ff3_1\n",
            "   - conda-libmamba-solver==25.3.0=pyhd8ed1ab_0\n",
            "\n",
            "\n",
            "  Package                         Version  Build               Channel         Size\n",
            "─────────────────────────────────────────────────────────────────────────────────────\n",
            "  Install:\n",
            "─────────────────────────────────────────────────────────────────────────────────────\n",
            "\n",
            "  + _libgcc_mutex                     0.1  conda_forge         conda-forge         \n",
            "  + _openmp_mutex                     4.5  2_gnu               conda-forge         \n",
            "  + archspec                        0.2.5  pyhd8ed1ab_0        conda-forge         \n",
            "  + boltons                        25.0.0  pyhd8ed1ab_0        conda-forge         \n",
            "  + brotli-python                   1.1.0  py312h2ec8cdc_3     conda-forge         \n",
            "  + bzip2                           1.0.8  h4bc722e_7          conda-forge         \n",
            "  + c-ares                         1.34.5  hb9d3cd8_0          conda-forge         \n",
            "  + ca-certificates             2025.7.14  hbd8a1cb_0          conda-forge         \n",
            "  + certifi                     2025.7.14  pyhd8ed1ab_0        conda-forge         \n",
            "  + cffi                           1.17.1  py312h06ac9bb_0     conda-forge         \n",
            "  + charset-normalizer              3.4.2  pyhd8ed1ab_0        conda-forge         \n",
            "  + colorama                        0.4.6  pyhd8ed1ab_1        conda-forge         \n",
            "  + conda                          25.3.1  py312h7900ff3_1     conda-forge         \n",
            "  + conda-libmamba-solver          25.3.0  pyhd8ed1ab_0        conda-forge         \n",
            "  + conda-package-handling          2.4.0  pyh7900ff3_2        conda-forge         \n",
            "  + conda-package-streaming        0.12.0  pyhd8ed1ab_0        conda-forge         \n",
            "  + cpp-expected                    1.1.0  hff21bea_1          conda-forge         \n",
            "  + distro                          1.9.0  pyhd8ed1ab_1        conda-forge         \n",
            "  + fmt                            11.1.4  h07f6e7f_1          conda-forge         \n",
            "  + frozendict                      2.4.6  py312h66e93f0_0     conda-forge         \n",
            "  + h2                              4.2.0  pyhd8ed1ab_0        conda-forge         \n",
            "  + hpack                           4.1.0  pyhd8ed1ab_0        conda-forge         \n",
            "  + hyperframe                      6.1.0  pyhd8ed1ab_0        conda-forge         \n",
            "  + icu                              75.1  he02047a_0          conda-forge         \n",
            "  + idna                             3.10  pyhd8ed1ab_1        conda-forge         \n",
            "  + jsonpatch                        1.33  pyhd8ed1ab_1        conda-forge         \n",
            "  + jsonpointer                     3.0.0  py312h7900ff3_1     conda-forge         \n",
            "  + keyutils                        1.6.1  h166bdaf_0          conda-forge         \n",
            "  + krb5                           1.21.3  h659f571_0          conda-forge         \n",
            "  + ld_impl_linux-64                 2.44  h1423503_1          conda-forge         \n",
            "  + libarchive                      3.7.7  h75ea233_4          conda-forge         \n",
            "  + libcurl                        8.14.1  h332b0f4_0          conda-forge         \n",
            "  + libedit                  3.1.20250104  pl5321h7949ede_0    conda-forge         \n",
            "  + libev                            4.33  hd590300_2          conda-forge         \n",
            "  + libexpat                        2.7.1  hecca717_0          conda-forge         \n",
            "  + libffi                          3.4.6  h2dba641_1          conda-forge         \n",
            "  + libgcc                         15.1.0  h767d61c_3          conda-forge         \n",
            "  + libgcc-ng                      15.1.0  h69a702a_3          conda-forge         \n",
            "  + libgomp                        15.1.0  h767d61c_3          conda-forge         \n",
            "  + libiconv                         1.18  h4ce23a2_1          conda-forge         \n",
            "  + liblzma                         5.8.1  hb9d3cd8_2          conda-forge         \n",
            "  + libmamba                        2.1.1  h430c389_0          conda-forge         \n",
            "  + libmambapy                      2.1.1  py312h07448e0_0     conda-forge         \n",
            "  + libnghttp2                     1.64.0  h161d5f1_0          conda-forge         \n",
            "  + libnsl                          2.0.1  hb9d3cd8_1          conda-forge         \n",
            "  + libsolv                        0.7.34  h9463b59_0          conda-forge         \n",
            "  + libsqlite                      3.50.3  hee844dc_1          conda-forge         \n",
            "  + libssh2                        1.11.1  hcf80075_0          conda-forge         \n",
            "  + libstdcxx                      15.1.0  h8f9b012_3          conda-forge         \n",
            "  + libstdcxx-ng                   15.1.0  h4852527_3          conda-forge         \n",
            "  + libuuid                        2.38.1  h0b41bf4_0          conda-forge         \n",
            "  + libxcrypt                      4.4.36  hd590300_1          conda-forge         \n",
            "  + libxml2                        2.13.8  h4bc477f_0          conda-forge         \n",
            "  + libzlib                         1.3.1  hb9d3cd8_2          conda-forge         \n",
            "  + lz4-c                          1.10.0  h5888daf_1          conda-forge         \n",
            "  + lzo                              2.10  hd590300_1001       conda-forge         \n",
            "  + mamba                           2.1.1  had4a41a_0          conda-forge         \n",
            "  + menuinst                        2.3.1  py312h7900ff3_0     conda-forge         \n",
            "  + ncurses                           6.5  h2d0b736_3          conda-forge         \n",
            "  + nlohmann_json                  3.11.3  he02047a_1          conda-forge         \n",
            "  + openssl                         3.5.1  h7b32b05_0          conda-forge         \n",
            "  + packaging                        25.0  pyh29332c3_1        conda-forge         \n",
            "  + pip                            25.1.1  pyh8b19718_0        conda-forge         \n",
            "  + platformdirs                    4.3.8  pyhe01879c_0        conda-forge         \n",
            "  + pluggy                          1.6.0  pyhd8ed1ab_0        conda-forge         \n",
            "  + pybind11-abi                        4  hd8ed1ab_3          conda-forge         \n",
            "  + pycosat                         0.6.6  py312h66e93f0_2     conda-forge         \n",
            "  + pycparser                        2.22  pyh29332c3_1        conda-forge         \n",
            "  + pysocks                         1.7.1  pyha55dd90_7        conda-forge         \n",
            "  + python                        3.12.11  h9e4cc4f_0_cpython  conda-forge         \n",
            "  + python_abi                       3.12  8_cp312             conda-forge         \n",
            "  + readline                          8.2  h8c095d6_2          conda-forge         \n",
            "  + reproc                   14.2.5.post0  hb9d3cd8_0          conda-forge         \n",
            "  + reproc-cpp               14.2.5.post0  h5888daf_0          conda-forge         \n",
            "  + requests                       2.32.4  pyhd8ed1ab_0        conda-forge         \n",
            "  + ruamel.yaml                   0.18.14  py312h66e93f0_0     conda-forge         \n",
            "  + ruamel.yaml.clib                0.2.8  py312h66e93f0_1     conda-forge         \n",
            "  + setuptools                     80.9.0  pyhff2d567_0        conda-forge         \n",
            "  + simdjson                       3.12.3  h84d6215_0          conda-forge         \n",
            "  + tk                             8.6.13  noxft_hd72426e_102  conda-forge         \n",
            "  + tqdm                           4.67.1  pyhd8ed1ab_1        conda-forge         \n",
            "  + truststore                     0.10.1  pyh29332c3_0        conda-forge         \n",
            "  + tzdata                          2025b  h78e105d_0          conda-forge         \n",
            "  + urllib3                         2.5.0  pyhd8ed1ab_0        conda-forge         \n",
            "  + wheel                          0.45.1  pyhd8ed1ab_1        conda-forge         \n",
            "  + yaml-cpp                        0.8.0  h3f2d84a_0          conda-forge         \n",
            "  + zstandard                      0.23.0  py312h66e93f0_2     conda-forge         \n",
            "  + zstd                            1.5.7  hb8e6e7a_2          conda-forge         \n",
            "\n",
            "  Summary:\n",
            "\n",
            "  Install: 88 packages\n",
            "\n",
            "  Total download: 0 B\n",
            "\n",
            "─────────────────────────────────────────────────────────────────────────────────────\n",
            "\n",
            "\n",
            "\n",
            "Transaction starting\n",
            "Linking _libgcc_mutex-0.1-conda_forge\n",
            "Linking ca-certificates-2025.7.14-hbd8a1cb_0\n",
            "Linking ld_impl_linux-64-2.44-h1423503_1\n",
            "Linking libgomp-15.1.0-h767d61c_3\n",
            "Linking pybind11-abi-4-hd8ed1ab_3\n",
            "Linking python_abi-3.12-8_cp312\n",
            "Linking tzdata-2025b-h78e105d_0\n",
            "Linking _openmp_mutex-4.5-2_gnu\n",
            "Linking libgcc-15.1.0-h767d61c_3\n",
            "Linking c-ares-1.34.5-hb9d3cd8_0\n",
            "Linking libexpat-2.7.1-hecca717_0\n",
            "Linking libffi-3.4.6-h2dba641_1\n",
            "Linking libgcc-ng-15.1.0-h69a702a_3\n",
            "Linking libiconv-1.18-h4ce23a2_1\n",
            "Linking liblzma-5.8.1-hb9d3cd8_2\n",
            "Linking libnsl-2.0.1-hb9d3cd8_1\n",
            "Linking libstdcxx-15.1.0-h8f9b012_3\n",
            "Linking libzlib-1.3.1-hb9d3cd8_2\n",
            "Linking ncurses-6.5-h2d0b736_3\n",
            "Linking openssl-3.5.1-h7b32b05_0\n",
            "Linking reproc-14.2.5.post0-hb9d3cd8_0\n",
            "Linking bzip2-1.0.8-h4bc722e_7\n",
            "Linking cpp-expected-1.1.0-hff21bea_1\n",
            "Linking fmt-11.1.4-h07f6e7f_1\n",
            "Linking keyutils-1.6.1-h166bdaf_0\n",
            "Linking libedit-3.1.20250104-pl5321h7949ede_0\n",
            "Linking libev-4.33-hd590300_2\n",
            "Linking libsolv-0.7.34-h9463b59_0\n",
            "Linking libssh2-1.11.1-hcf80075_0\n",
            "Linking libstdcxx-ng-15.1.0-h4852527_3\n",
            "Linking libuuid-2.38.1-h0b41bf4_0\n",
            "Linking libxcrypt-4.4.36-hd590300_1\n",
            "Linking lz4-c-1.10.0-h5888daf_1\n",
            "Linking lzo-2.10-hd590300_1001\n",
            "Linking readline-8.2-h8c095d6_2\n",
            "Linking reproc-cpp-14.2.5.post0-h5888daf_0\n",
            "Linking simdjson-3.12.3-h84d6215_0\n",
            "Linking tk-8.6.13-noxft_hd72426e_102\n",
            "Linking yaml-cpp-0.8.0-h3f2d84a_0\n",
            "Linking zstd-1.5.7-hb8e6e7a_2\n",
            "Linking icu-75.1-he02047a_0\n",
            "Linking krb5-1.21.3-h659f571_0\n",
            "Linking libnghttp2-1.64.0-h161d5f1_0\n",
            "Linking nlohmann_json-3.11.3-he02047a_1\n",
            "Linking libcurl-8.14.1-h332b0f4_0\n",
            "Linking libsqlite-3.50.3-hee844dc_1\n",
            "Linking libxml2-2.13.8-h4bc477f_0\n",
            "Linking libarchive-3.7.7-h75ea233_4\n",
            "Linking python-3.12.11-h9e4cc4f_0_cpython\n",
            "Linking libmamba-2.1.1-h430c389_0\n",
            "Linking menuinst-2.3.1-py312h7900ff3_0\n",
            "Linking archspec-0.2.5-pyhd8ed1ab_0\n",
            "Linking boltons-25.0.0-pyhd8ed1ab_0\n",
            "Linking brotli-python-1.1.0-py312h2ec8cdc_3\n",
            "Linking certifi-2025.7.14-pyhd8ed1ab_0\n",
            "Linking charset-normalizer-3.4.2-pyhd8ed1ab_0\n",
            "Linking colorama-0.4.6-pyhd8ed1ab_1\n",
            "Linking distro-1.9.0-pyhd8ed1ab_1\n",
            "Linking frozendict-2.4.6-py312h66e93f0_0\n",
            "Linking hpack-4.1.0-pyhd8ed1ab_0\n",
            "Linking hyperframe-6.1.0-pyhd8ed1ab_0\n",
            "Linking idna-3.10-pyhd8ed1ab_1\n",
            "Linking jsonpointer-3.0.0-py312h7900ff3_1\n",
            "Linking libmambapy-2.1.1-py312h07448e0_0\n",
            "Linking mamba-2.1.1-had4a41a_0\n",
            "Linking packaging-25.0-pyh29332c3_1\n",
            "Linking platformdirs-4.3.8-pyhe01879c_0\n",
            "Linking pluggy-1.6.0-pyhd8ed1ab_0\n",
            "Linking pycosat-0.6.6-py312h66e93f0_2\n",
            "Linking pycparser-2.22-pyh29332c3_1\n",
            "Linking pysocks-1.7.1-pyha55dd90_7\n",
            "Linking ruamel.yaml.clib-0.2.8-py312h66e93f0_1\n",
            "Linking setuptools-80.9.0-pyhff2d567_0\n",
            "Linking truststore-0.10.1-pyh29332c3_0\n",
            "Linking wheel-0.45.1-pyhd8ed1ab_1\n",
            "Linking cffi-1.17.1-py312h06ac9bb_0\n",
            "Linking h2-4.2.0-pyhd8ed1ab_0\n",
            "Linking jsonpatch-1.33-pyhd8ed1ab_1\n",
            "Linking pip-25.1.1-pyh8b19718_0\n",
            "Linking ruamel.yaml-0.18.14-py312h66e93f0_0\n",
            "Linking tqdm-4.67.1-pyhd8ed1ab_1\n",
            "Linking zstandard-0.23.0-py312h66e93f0_2\n",
            "Linking conda-package-streaming-0.12.0-pyhd8ed1ab_0\n",
            "Linking urllib3-2.5.0-pyhd8ed1ab_0\n",
            "Linking requests-2.32.4-pyhd8ed1ab_0\n",
            "Linking conda-package-handling-2.4.0-pyh7900ff3_2\n",
            "Linking conda-25.3.1-py312h7900ff3_1\n",
            "Linking conda-libmamba-solver-25.3.0-pyhd8ed1ab_0\n",
            "\n",
            "Transaction finished\n",
            "\n",
            "installation finished.\n",
            "WARNING:\n",
            "    You currently have a PYTHONPATH environment variable set. This may cause\n",
            "    unexpected behavior when running the Python interpreter in Miniforge3.\n",
            "    For best results, please verify that your PYTHONPATH only points to\n",
            "    directories of packages that are compatible with the Python interpreter\n",
            "    in Miniforge3: /usr/local\n",
            "Running `shell init`, which:\n",
            " - modifies RC file: \"/root/.bashrc\"\n",
            " - generates config for root prefix: \u001b[1m\"/root/.local/share/mamba\"\u001b[0m\n",
            " - sets mamba executable to: \u001b[1m\"/usr/local/bin/mamba\"\u001b[0m\n",
            "The following has been added in your \"/root/.bashrc\" file\n",
            "\n",
            "# >>> mamba initialize >>>\n",
            "# !! Contents within this block are managed by 'mamba shell init' !!\n",
            "export MAMBA_EXE='/usr/local/bin/mamba';\n",
            "export MAMBA_ROOT_PREFIX='/root/.local/share/mamba';\n",
            "__mamba_setup=\"$(\"$MAMBA_EXE\" shell hook --shell bash --root-prefix \"$MAMBA_ROOT_PREFIX\" 2> /dev/null)\"\n",
            "if [ $? -eq 0 ]; then\n",
            "    eval \"$__mamba_setup\"\n",
            "else\n",
            "    alias mamba=\"$MAMBA_EXE\"  # Fallback on help from mamba activate\n",
            "fi\n",
            "unset __mamba_setup\n",
            "# <<< mamba initialize <<<\n",
            "\n"
          ]
        },
        {
          "output_type": "stream",
          "name": "stderr",
          "text": [
            "warning  libmamba [python-3.12.11-h9e4cc4f_0_cpython] The following files were already present in the environment:\n",
            "    - bin/python\n",
            "warning  libmamba [charset-normalizer-3.4.2-pyhd8ed1ab_0] The following files were already present in the environment:\n",
            "    - bin/normalizer\n",
            "warning  libmamba [distro-1.9.0-pyhd8ed1ab_1] The following files were already present in the environment:\n",
            "    - bin/distro\n",
            "warning  libmamba [jsonpointer-3.0.0-py312h7900ff3_1] The following files were already present in the environment:\n",
            "    - bin/jsonpointer\n",
            "warning  libmamba [wheel-0.45.1-pyhd8ed1ab_1] The following files were already present in the environment:\n",
            "    - bin/wheel\n",
            "warning  libmamba [jsonpatch-1.33-pyhd8ed1ab_1] The following files were already present in the environment:\n",
            "    - bin/jsondiff\n",
            "    - bin/jsonpatch\n",
            "warning  libmamba [pip-25.1.1-pyh8b19718_0] The following files were already present in the environment:\n",
            "    - bin/pip\n",
            "    - bin/pip3\n",
            "warning  libmamba [tqdm-4.67.1-pyhd8ed1ab_1] The following files were already present in the environment:\n",
            "    - bin/tqdm\n",
            "The following argument was not expected: init\n",
            "Run with --help for more information.\n"
          ]
        }
      ],
      "source": [
        "%%bash\n",
        "# Instalar miniconda e mamba\n",
        "wget --quiet https://github.com/conda-forge/miniforge/releases/latest/download/Miniforge3-Linux-x86_64.sh\n",
        "bash Miniforge3-Linux-x86_64.sh -b -p /usr/local -u\n",
        "mamba init\n",
        "mamba config set auto_activate_base true\n",
        "mamba shell init --shell bash --root-prefix=~/.local/share/mamba\n",
        "rm Miniforge3-Linux-x86_64.sh"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": null,
      "metadata": {
        "collapsed": true,
        "id": "6o9JEhp6F0JD",
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "outputId": "721e91ff-2fdc-4dc9-f8ae-8e50b4d92bd6"
      },
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "\n",
            "Pinned packages:\n",
            "\n",
            "  - python=3.12\n",
            "\n",
            "\n",
            "Transaction\n",
            "\n",
            "  Prefix: /usr/local\n",
            "\n",
            "  Updating specs:\n",
            "\n",
            "   - fastqc\n",
            "   - bwa\n",
            "   - gatk4\n",
            "   - cutadapt\n",
            "   - samtools\n",
            "   - bedtools\n",
            "   - ensembl-vep=115\n",
            "\n",
            "\n",
            "  Package                                           Version  Build                  Channel           Size\n",
            "────────────────────────────────────────────────────────────────────────────────────────────────────────────\n",
            "  Install:\n",
            "────────────────────────────────────────────────────────────────────────────────────────────────────────────\n",
            "\n",
            "  + alsa-lib                                         1.2.11  hd590300_1             conda-forge      555kB\n",
            "  + argtable2                                          2.13  hd590300_1004          conda-forge        2MB\n",
            "  + bedtools                                         2.31.1  hf5e1c6e_2             bioconda           2MB\n",
            "  + binutils_impl_linux-64                             2.44  h4bf12b8_1             conda-forge        4MB\n",
            "  + binutils_linux-64                                  2.44  h4852527_1             conda-forge       36kB\n",
            "  + blast-legacy                                     2.2.26  hf7ff83a_5             bioconda          46MB\n",
            "  + bwa                                              0.7.18  he4a0461_1             bioconda         196kB\n",
            "  + cairo                                            1.18.0  h3faef2a_0             conda-forge      982kB\n",
            "  + clustalo                                          1.2.4  h503566f_10            bioconda         219kB\n",
            "  + clustalw                                            2.1  h9948957_12            bioconda         385kB\n",
            "  + consan                                              1.2  h7b50bb2_7             bioconda           1MB\n",
            "  + cutadapt                                            5.2  py312h0fa9677_0        bioconda         318kB\n",
            "  + cyrus-sasl                                       2.1.28  hd9c7081_0             conda-forge      210kB\n",
            "  + dendropy                                          5.0.8  pyhdfd78af_1           bioconda         335kB\n",
            "  + dialign-tx                                        1.0.2  h7b50bb2_8             bioconda          53kB\n",
            "  + dnaio                                             1.2.2  py312hf67a6ed_0        bioconda         153kB\n",
            "  + dnspython                                         2.8.0  pyhcf101f3_0           conda-forge      196kB\n",
            "  + ensembl-vep                                       115.2  pl5321h2a3209d_1       bioconda           4MB\n",
            "  + expat                                             2.7.1  hecca717_0             conda-forge      141kB\n",
            "  + famsa                                             2.2.3  h9948957_1             bioconda         527kB\n",
            "  + fastqc                                           0.12.1  hdfd78af_0             bioconda          12MB\n",
            "  + fasttree                                          2.2.0  h7b50bb2_0             bioconda         205kB\n",
            "  + font-ttf-dejavu-sans-mono                          2.37  hab24e00_0             conda-forge      397kB\n",
            "  + font-ttf-inconsolata                              3.000  h77eed37_0             conda-forge       97kB\n",
            "  + font-ttf-source-code-pro                          2.038  h77eed37_0             conda-forge      701kB\n",
            "  + font-ttf-ubuntu                                    0.83  h77eed37_3             conda-forge        2MB\n",
            "  + fontconfig                                       2.14.2  h14ed4e7_0             conda-forge      272kB\n",
            "  + fonts-conda-ecosystem                                 1  0                      conda-forge        4kB\n",
            "  + fonts-conda-forge                                     1  0                      conda-forge        4kB\n",
            "  + freetype                                         2.12.1  h267a509_2             conda-forge      635kB\n",
            "  + gatk4                                           4.6.2.0  py310hdfd78af_0        bioconda         399MB\n",
            "  + gawk                                              5.3.1  hcd3d067_0             conda-forge        1MB\n",
            "  + gcc_impl_linux-64                                13.4.0  h69c5793_7             conda-forge       67MB\n",
            "  + gcc_linux-64                                     13.4.0  h0a5b801_12            conda-forge       28kB\n",
            "  + giflib                                            5.2.2  hd590300_0             conda-forge       77kB\n",
            "  + gmp                                               6.3.0  hac33072_2             conda-forge      460kB\n",
            "  + gnutls                                            3.8.9  h86084c0_1             conda-forge        2MB\n",
            "  + graphite2                                        1.3.14  hecca717_2             conda-forge      100kB\n",
            "  + gsl                                                 2.7  he838d99_0             conda-forge        3MB\n",
            "  + harfbuzz                                          9.0.0  hfac3d4d_0             conda-forge        2MB\n",
            "  + hmmer                                               3.4  hb6cb901_4             bioconda          12MB\n",
            "  + htslib                                             1.21  h5efdd21_0             bioconda           3MB\n",
            "  + isa-l                                            2.31.1  hb9d3cd8_1             conda-forge      157kB\n",
            "  + kalign2                                            2.04  h7b50bb2_8             bioconda         106kB\n",
            "  + kernel-headers_linux-64                          5.14.0  he073ed8_2             conda-forge        1MB\n",
            "  + lcms2                                              2.16  hb7c19ff_0             conda-forge      245kB\n",
            "  + lerc                                              4.0.0  h0aef613_1             conda-forge      264kB\n",
            "  + libabseil                                    20240116.2  cxx17_he02047a_1       conda-forge        1MB\n",
            "  + libasprintf                                      0.25.1  h3f43e3d_1             conda-forge       54kB\n",
            "  + libblas                                           3.9.0  38_h4a7cf45_openblas   conda-forge       18kB\n",
            "  + libcblas                                          3.9.0  38_h0358290_openblas   conda-forge       18kB\n",
            "  + libcups                                           2.3.3  h4637d8d_4             conda-forge        5MB\n",
            "  + libdb                                            6.2.32  h9c3ff4c_0             conda-forge       24MB\n",
            "  + libdeflate                                         1.20  hd590300_0             conda-forge       72kB\n",
            "  + libevent                                         2.1.12  hf998b51_1             conda-forge      427kB\n",
            "  + libgcc-devel_linux-64                            13.4.0  hba01cd7_107           conda-forge        3MB\n",
            "  + libgettextpo                                     0.25.1  h3f43e3d_1             conda-forge      188kB\n",
            "  + libgfortran                                      15.1.0  h69a702a_5             conda-forge       29kB\n",
            "  + libgfortran-ng                                   15.1.0  h69a702a_5             conda-forge       29kB\n",
            "  + libgfortran5                                     15.1.0  hcea5267_5             conda-forge        2MB\n",
            "  + libglib                                          2.80.2  hf974151_0             conda-forge        4MB\n",
            "  + libidn2                                           2.3.8  hfac485b_1             conda-forge      139kB\n",
            "  + libjpeg-turbo                                     3.1.0  hb9d3cd8_0             conda-forge      629kB\n",
            "  + liblzma-devel                                     5.8.1  hb9d3cd8_2             conda-forge      440kB\n",
            "  + libntlm                                             1.8  hb9d3cd8_0             conda-forge       33kB\n",
            "  + libopenblas                                      0.3.30  pthreads_h94d23a6_3    conda-forge        6MB\n",
            "  + libpng                                           1.6.43  h2797004_0             conda-forge      288kB\n",
            "  + libprotobuf                                      4.25.3  h08a7969_0             conda-forge        3MB\n",
            "  + libsanitizer                                     13.4.0  h14bf0c3_7             conda-forge        5MB\n",
            "  + libtasn1                                         4.20.0  hb03c661_1             conda-forge      118kB\n",
            "  + libtiff                                           4.6.0  h1dd3fc0_3             conda-forge      283kB\n",
            "  + libunistring                                     0.9.10  h7f98852_0             conda-forge        1MB\n",
            "  + libwebp-base                                      1.6.0  hd42ef1d_0             conda-forge      429kB\n",
            "  + libxcb                                             1.15  h0b41bf4_0             conda-forge      384kB\n",
            "  + mafft                                             7.526  h4bc722e_0             conda-forge        3MB\n",
            "  + mpfr                                              4.2.1  h90cbb55_3             conda-forge      635kB\n",
            "  + mpi                                                 1.0  openmpi                conda-forge        4kB\n",
            "  + muscle                                         3.8.1551  h9948957_9             bioconda         274kB\n",
            "  + mustang                                           3.2.4  h9948957_0             bioconda         115kB\n",
            "  + mysql                                             8.3.0  h27aab58_4             conda-forge      348kB\n",
            "  + mysql-client                                      8.3.0  h545f5f4_4             conda-forge        5MB\n",
            "  + mysql-common                                      8.3.0  hf1915f5_4             conda-forge      785kB\n",
            "  + mysql-connector-c                                6.1.11  h659d440_1008          conda-forge        1MB\n",
            "  + mysql-devel                                       8.3.0  hf1915f5_4             conda-forge        2MB\n",
            "  + mysql-libs                                        8.3.0  hca2cd23_4             conda-forge        2MB\n",
            "  + mysql-server                                      8.3.0  h27b9c54_4             conda-forge       18MB\n",
            "  + nettle                                           3.10.1  h4a9d5aa_0             conda-forge        1MB\n",
            "  + openjdk                                         17.0.11  h24d6bf4_0             conda-forge      170MB\n",
            "  + openmpi                                           4.1.6  hc5af2df_101           conda-forge        4MB\n",
            "  + p11-kit                                          0.24.1  hc5aa10d_0             conda-forge        5MB\n",
            "  + paml                                             4.10.9  h7b50bb2_1             bioconda         679kB\n",
            "  + pasta                                             1.9.2  py312hccd54bf_5        bioconda           1MB\n",
            "  + pbzip2                                           1.1.13  h1fcc475_2             conda-forge       42kB\n",
            "  + pcre                                               8.45  h9c3ff4c_0             conda-forge      259kB\n",
            "  + pcre2                                             10.43  hcad00b1_0             conda-forge      951kB\n",
            "  + perl                                             5.32.1  7_hd590300_perl5       conda-forge       13MB\n",
            "  + perl-algorithm-diff                              1.1903  pl5321hdfd78af_3       bioconda          37kB\n",
            "  + perl-b-cow                                        0.007  pl5321hb9d3cd8_1       conda-forge       21kB\n",
            "  + perl-base                                          2.23  pl5321hdfd78af_2       bioconda          12kB\n",
            "  + perl-bio-asn1-entrezgene                           1.73  pl5321hdfd78af_3       bioconda          25kB\n",
            "  + perl-bio-bigfile                                   1.07  pl5321h41f7678_5       bioconda         832kB\n",
            "  + perl-bio-coordinate                            1.007001  pl5321hdfd78af_3       bioconda          37kB\n",
            "  + perl-bio-db-embl                                  1.7.4  pl5321hdfd78af_0       bioconda          11kB\n",
            "  + perl-bio-db-hts                                    3.01  pl5321h577a1d6_12      bioconda         119kB\n",
            "  + perl-bio-db-refseq                                1.7.4  pl5321hdfd78af_0       bioconda          12kB\n",
            "  + perl-bio-db-swissprot                             1.7.4  pl5321hdfd78af_0       bioconda          16kB\n",
            "  + perl-bio-featureio                              1.6.905  pl5321hdfd78af_4       bioconda          42kB\n",
            "  + perl-bio-procedural                               1.7.4  pl5321hdfd78af_1       bioconda          17kB\n",
            "  + perl-bio-samtools                                  1.43  pl5321he4a0461_4       bioconda         153kB\n",
            "  + perl-bio-searchio-hmmer                           1.7.3  pl5321hdfd78af_0       bioconda          59kB\n",
            "  + perl-bio-tools-phylo-paml                         1.7.3  pl5321hdfd78af_3       bioconda          59kB\n",
            "  + perl-bio-tools-run-alignment-clustalw             1.7.4  pl5321hdfd78af_3       bioconda          26kB\n",
            "  + perl-bio-tools-run-alignment-tcoffee              1.7.4  pl5321hdfd78af_5       bioconda          26kB\n",
            "  + perl-bio-tools-run-remoteblast                    1.7.3  pl5321hdfd78af_0       bioconda          20kB\n",
            "  + perl-bioperl                                      1.7.8  hdfd78af_1             bioconda           6kB\n",
            "  + perl-bioperl-core                                 1.7.8  pl5321hdfd78af_1       bioconda           2MB\n",
            "  + perl-bioperl-run                               1.007003  pl5321hdfd78af_0       bioconda         348kB\n",
            "  + perl-business-isbn                                3.007  pl5321hdfd78af_0       bioconda          18kB\n",
            "  + perl-business-isbn-data                    20210112.006  pl5321hdfd78af_0       bioconda          21kB\n",
            "  + perl-capture-tiny                                  0.48  pl5321hdfd78af_2       bioconda          19kB\n",
            "  + perl-carp                                          1.38  pl5321hdfd78af_4       bioconda          17kB\n",
            "  + perl-class-data-inheritable                        0.09  pl5321hdfd78af_0       bioconda          10kB\n",
            "  + perl-clone                                         0.46  pl5321hec16e2b_0       bioconda          16kB\n",
            "  + perl-common-sense                                  3.75  pl5321hdfd78af_0       bioconda          14kB\n",
            "  + perl-compress-raw-bzip2                           2.214  pl5321hda65f42_0       conda-forge       56kB\n",
            "  + perl-compress-raw-zlib                            2.214  pl5321h4dac143_0       conda-forge       81kB\n",
            "  + perl-constant                                      1.33  pl5321hdfd78af_2       bioconda          12kB\n",
            "  + perl-data-dump                                     1.25  pl5321h7b50bb2_2       bioconda          25kB\n",
            "  + perl-data-dumper                                  2.183  pl5321hec16e2b_1       bioconda          38kB\n",
            "  + perl-db_file                                      1.858  pl5321hb9d3cd8_0       conda-forge       64kB\n",
            "  + perl-dbd-mysql                                    4.050  pl5321h9948957_3       bioconda         255kB\n",
            "  + perl-dbd-sqlite                                    1.76  pl5321h7b50bb2_2       bioconda           3MB\n",
            "  + perl-dbi                                          1.643  pl5321hec16e2b_1       bioconda         619kB\n",
            "  + perl-devel-checklib                                1.16  pl5321h7b50bb2_1       bioconda          19kB\n",
            "  + perl-devel-stacktrace                              2.04  pl5321hdfd78af_1       bioconda          17kB\n",
            "  + perl-digest-hmac                                   1.05  pl5321hdfd78af_0       bioconda          13kB\n",
            "  + perl-digest-md5                                    2.58  pl5321hec16e2b_1       bioconda          20kB\n",
            "  + perl-encode                                        3.19  pl5321hec16e2b_1       bioconda           2MB\n",
            "  + perl-encode-locale                                 1.05  pl5321hdfd78af_7       bioconda          14kB\n",
            "  + perl-exception-class                               1.45  pl5321hdfd78af_0       bioconda          22kB\n",
            "  + perl-exporter                                      5.72  pl5321hdfd78af_2       bioconda          16kB\n",
            "  + perl-exporter-tiny                             1.002002  pl5321hdfd78af_0       bioconda          25kB\n",
            "  + perl-extutils-cbuilder                         0.280230  pl5321hdfd78af_2       bioconda          23kB\n",
            "  + perl-extutils-cppguess                             0.26  pl5321h9948957_4       bioconda          16kB\n",
            "  + perl-extutils-makemaker                            7.70  pl5321hd8ed1ab_0       conda-forge      157kB\n",
            "  + perl-extutils-manifest                             1.75  pl5321hdfd78af_0       bioconda          16kB\n",
            "  + perl-extutils-parsexs                              3.60  pl5321hdfd78af_0       bioconda         143kB\n",
            "  + perl-file-listing                                  6.16  pl5321hdfd78af_0       bioconda          15kB\n",
            "  + perl-file-path                                     2.18  pl5321hd8ed1ab_0       conda-forge       22kB\n",
            "  + perl-file-slurper                                 0.014  pl5321hdfd78af_0       bioconda          12kB\n",
            "  + perl-file-sort                                     1.01  pl5321hdfd78af_3       bioconda          21kB\n",
            "  + perl-file-spec                                  3.48_01  pl5321hdfd78af_2       bioconda           6kB\n",
            "  + perl-file-temp                                   0.2304  pl5321hd8ed1ab_0       conda-forge       32kB\n",
            "  + perl-getopt-long                                   2.58  pl5321hdfd78af_0       bioconda          34kB\n",
            "  + perl-html-parser                                   3.81  pl5321h4ac6f70_1       bioconda          70kB\n",
            "  + perl-html-tagset                                   3.24  pl5321hdfd78af_0       bioconda          15kB\n",
            "  + perl-http-cookiejar-lwp                           0.014  pl5321hdfd78af_0       bioconda          18kB\n",
            "  + perl-http-cookies                                  6.11  pl5321hdfd78af_0       bioconda          24kB\n",
            "  + perl-http-daemon                                   6.16  pl5321hdfd78af_0       bioconda          22kB\n",
            "  + perl-http-date                                     6.06  pl5321hdfd78af_0       bioconda          15kB\n",
            "  + perl-http-message                                  7.01  pl5321hdfd78af_0       bioconda          59kB\n",
            "  + perl-http-negotiate                                6.01  pl5321hdfd78af_4       bioconda          15kB\n",
            "  + perl-inc-latest                                   0.500  pl5321ha770c72_0       conda-forge       16kB\n",
            "  + perl-io-compress                                  2.213  pl5321h503566f_0       bioconda          87kB\n",
            "  + perl-io-html                                      1.004  pl5321hdfd78af_0       bioconda          16kB\n",
            "  + perl-io-socket-ssl                                2.074  pl5321hdfd78af_0       bioconda         166kB\n",
            "  + perl-io-string                                     1.08  pl5321hdfd78af_4       bioconda          12kB\n",
            "  + perl-io-tty                                        1.16  pl5321hec16e2b_1       bioconda          33kB\n",
            "  + perl-ipc-cmd                                       1.04  pl5321hdfd78af_1       bioconda          27kB\n",
            "  + perl-ipc-run                                 20250809.0  pl5321hdfd78af_0       bioconda          85kB\n",
            "  + perl-json                                          4.10  pl5321hdfd78af_1       bioconda          58kB\n",
            "  + perl-json-xs                                       4.04  pl5321h9948957_0       bioconda          71kB\n",
            "  + perl-libwww-perl                                   6.81  pl5321hdfd78af_0       bioconda         103kB\n",
            "  + perl-libxml-perl                                   0.08  pl5321hdfd78af_3       bioconda          41kB\n",
            "  + perl-list-moreutils                               0.430  pl5321hdfd78af_0       bioconda          32kB\n",
            "  + perl-list-moreutils-xs                            0.430  pl5321h7b50bb2_5       bioconda          52kB\n",
            "  + perl-locale-maketext-simple                        0.21  pl5321hdfd78af_3       bioconda          11kB\n",
            "  + perl-lwp-mediatypes                                6.04  pl5321hdfd78af_1       bioconda          24kB\n",
            "  + perl-lwp-protocol-https                            6.14  pl5321hdfd78af_1       bioconda          13kB\n",
            "  + perl-mime-base64                                   3.16  pl5321hec16e2b_2       bioconda          18kB\n",
            "  + perl-module-build                                0.4234  pl5321ha770c72_1       conda-forge      137kB\n",
            "  + perl-module-corelist                         5.20251022  pl5321hdfd78af_0       bioconda          83kB\n",
            "  + perl-module-load                                   0.34  pl5321hdfd78af_0       bioconda           6kB\n",
            "  + perl-module-load-conditional                       0.68  pl5321hdfd78af_3       bioconda          12kB\n",
            "  + perl-module-metadata                           1.000038  pl5321hdfd78af_0       bioconda          19kB\n",
            "  + perl-mozilla-ca                                20250602  pl5321hdfd78af_0       bioconda         133kB\n",
            "  + perl-net-http                                      6.24  pl5321hdfd78af_0       bioconda          23kB\n",
            "  + perl-net-ssleay                                    1.92  pl5321hf14f497_1       conda-forge      311kB\n",
            "  + perl-ntlm                                          1.09  pl5321hdfd78af_5       bioconda          16kB\n",
            "  + perl-params-check                                  0.38  pl5321hdfd78af_2       bioconda           6kB\n",
            "  + perl-parent                                       0.236  pl5321hdfd78af_2       bioconda           8kB\n",
            "  + perl-perl-ostype                                  1.010  pl5321hdfd78af_2       bioconda           9kB\n",
            "  + perl-perlio-gzip                                   0.20  pl5321he4a0461_6       bioconda          19kB\n",
            "  + perl-scalar-list-utils                             1.62  pl5321hec16e2b_1       bioconda          45kB\n",
            "  + perl-sereal                                       5.004  pl5321hdfd78af_0       bioconda          14kB\n",
            "  + perl-sereal-decoder                               5.004  pl5321h7b50bb2_0       bioconda         177kB\n",
            "  + perl-sereal-encoder                               5.004  pl5321h7b50bb2_0       bioconda         308kB\n",
            "  + perl-set-intervaltree                              0.12  pl5321h503566f_6       bioconda          32kB\n",
            "  + perl-socket                                       2.027  pl5321h5c03b87_6       bioconda          35kB\n",
            "  + perl-sub-uplevel                                 0.2800  pl5321hec16e2b_4       bioconda          14kB\n",
            "  + perl-test-deep                                    1.130  pl5321hdfd78af_0       bioconda          38kB\n",
            "  + perl-test-differences                              0.69  pl5321hdfd78af_0       bioconda          19kB\n",
            "  + perl-test-exception                                0.43  pl5321hdfd78af_3       bioconda          15kB\n",
            "  + perl-test-harness                                  3.44  pl5321hdfd78af_0       bioconda          87kB\n",
            "  + perl-test-most                                     0.38  pl5321hdfd78af_0       bioconda          21kB\n",
            "  + perl-test-needs                                0.002009  pl5321hdfd78af_0       bioconda          12kB\n",
            "  + perl-test-requiresinternet                         0.05  pl5321hdfd78af_1       bioconda          10kB\n",
            "  + perl-test-warn                                     0.36  pl5321hdfd78af_2       bioconda          15kB\n",
            "  + perl-text-csv                                      2.01  pl5321hdfd78af_0       bioconda          72kB\n",
            "  + perl-text-diff                                     1.45  pl5321hdfd78af_1       bioconda          23kB\n",
            "  + perl-text-parsewords                               3.31  pl5321hdfd78af_0       bioconda          10kB\n",
            "  + perl-time-hires                                  1.9764  pl5321h7b50bb2_6       bioconda          28kB\n",
            "  + perl-time-local                                    1.35  pl5321hdfd78af_0       bioconda          14kB\n",
            "  + perl-timedate                                      2.33  pl5321hdfd78af_2       bioconda          32kB\n",
            "  + perl-tree-dag_node                                 1.35  pl5321hdfd78af_0       bioconda          45kB\n",
            "  + perl-try-tiny                                      0.31  pl5321hdfd78af_1       bioconda          20kB\n",
            "  + perl-types-serialiser                              1.01  pl5321hdfd78af_0       bioconda          13kB\n",
            "  + perl-uri                                           5.12  pl5321hdfd78af_0       bioconda          61kB\n",
            "  + perl-url-encode                                    0.03  pl5321h9ee0642_1       bioconda          13kB\n",
            "  + perl-version                                     0.9924  pl5321hec16e2b_2       bioconda          25kB\n",
            "  + perl-www-robotrules                                6.02  pl5321hdfd78af_4       bioconda          15kB\n",
            "  + perl-xml-dom                                       1.46  pl5321hdfd78af_1       bioconda          60kB\n",
            "  + perl-xml-dom-xpath                                 0.14  pl5321hdfd78af_2       bioconda          12kB\n",
            "  + perl-xml-parser                                 2.44_01  pl5321hf2e2c51_1004    conda-forge      150kB\n",
            "  + perl-xml-regexp                                    0.04  pl5321hdfd78af_3       bioconda          10kB\n",
            "  + perl-xml-xpathengine                               0.14  pl5321hdfd78af_3       bioconda          29kB\n",
            "  + phylip                                            3.697  h7b50bb2_3             bioconda           2MB\n",
            "  + pigz                                                2.8  h2797004_0             conda-forge       72kB\n",
            "  + pixman                                           0.46.4  h54a6638_1             conda-forge      451kB\n",
            "  + poa                                                 2.0  h7b50bb2_6             bioconda          82kB\n",
            "  + prank                                            170427  h9948957_1             bioconda         410kB\n",
            "  + probcons                                           1.12  h8b12597_1             bioconda         128kB\n",
            "  + probconsrna                                        1.10  h9f5acd7_4             bioconda         105kB\n",
            "  + proda                                               1.0  h503566f_7             bioconda         492kB\n",
            "  + pthread-stubs                                       0.4  hb9d3cd8_1002          conda-forge        8kB\n",
            "  + pymongo                                          4.15.3  py312h1289d80_0        conda-forge        2MB\n",
            "  + python-isal                                       1.8.0  py312h4c3975b_0        conda-forge       68kB\n",
            "  + python-zlib-ng                                    1.0.0  py312hd86e735_0        conda-forge       58kB\n",
            "  + raxml                                            8.2.13  h7b50bb2_3             bioconda           3MB\n",
            "  + samtools                                           1.21  h50ea8bc_0             bioconda         473kB\n",
            "  + sap                                               1.1.3  h7b50bb2_5             bioconda          51kB\n",
            "  + sniffio                                           1.3.1  pyhd8ed1ab_1           conda-forge       15kB\n",
            "  + sysroot_linux-64                                   2.34  h087de78_2             conda-forge       40MB\n",
            "  + t-coffee                               13.46.2.7c9e712d  pl5321hb2a3317_0       bioconda           2MB\n",
            "  + tmalign                                        20170708  h8d36177_7             bioconda          89kB\n",
            "  + unzip                                               6.0  hb03c661_4             conda-forge      147kB\n",
            "  + viennarna                                         2.7.0  py312pl5321h9f0f9fd_0  bioconda          32MB\n",
            "  + xopen                                             2.0.2  pyh707e725_2           conda-forge       22kB\n",
            "  + xorg-fixesproto                                     5.0  hb9d3cd8_1003          conda-forge       11kB\n",
            "  + xorg-inputproto                                   2.3.2  hb9d3cd8_1003          conda-forge       22kB\n",
            "  + xorg-kbproto                                      1.0.7  hb9d3cd8_1003          conda-forge       30kB\n",
            "  + xorg-libice                                       1.1.2  hb9d3cd8_0             conda-forge       59kB\n",
            "  + xorg-libsm                                        1.2.6  he73a12e_0             conda-forge       28kB\n",
            "  + xorg-libx11                                       1.8.9  h8ee46fc_0             conda-forge      828kB\n",
            "  + xorg-libxau                                      1.0.12  hb9d3cd8_0             conda-forge       15kB\n",
            "  + xorg-libxdmcp                                     1.1.5  hb9d3cd8_0             conda-forge       20kB\n",
            "  + xorg-libxext                                      1.3.4  h0b41bf4_2             conda-forge       50kB\n",
            "  + xorg-libxfixes                                    5.0.3  h7f98852_1004          conda-forge       18kB\n",
            "  + xorg-libxi                                       1.7.10  h4bc722e_1             conda-forge       47kB\n",
            "  + xorg-libxrender                                  0.9.11  hd590300_0             conda-forge       38kB\n",
            "  + xorg-libxt                                        1.3.0  hd590300_1             conda-forge      379kB\n",
            "  + xorg-libxtst                                      1.2.5  h4bc722e_0             conda-forge       33kB\n",
            "  + xorg-recordproto                                 1.14.2  hb9d3cd8_1003          conda-forge       10kB\n",
            "  + xorg-renderproto                                 0.11.1  hb9d3cd8_1003          conda-forge       12kB\n",
            "  + xorg-xextproto                                    7.3.0  hb9d3cd8_1004          conda-forge       31kB\n",
            "  + xorg-xproto                                      7.0.31  hb9d3cd8_1008          conda-forge       73kB\n",
            "  + xz                                                5.8.1  hbcc6ac9_2             conda-forge       24kB\n",
            "  + xz-gpl-tools                                      5.8.1  hbcc6ac9_2             conda-forge       34kB\n",
            "  + xz-tools                                          5.8.1  hb9d3cd8_2             conda-forge       96kB\n",
            "  + zlib                                             1.2.13  h4ab18f5_6             conda-forge       93kB\n",
            "  + zlib-ng                                           2.2.5  hde8ca8f_0             conda-forge      111kB\n",
            "\n",
            "  Change:\n",
            "────────────────────────────────────────────────────────────────────────────────────────────────────────────\n",
            "\n",
            "  - tk                                               8.6.13  noxft_hd72426e_102     conda-forge     Cached\n",
            "  + tk                                               8.6.13  noxft_h4845f30_101     conda-forge        3MB\n",
            "\n",
            "  Downgrade:\n",
            "────────────────────────────────────────────────────────────────────────────────────────────────────────────\n",
            "\n",
            "  - conda                                            25.3.1  py312h7900ff3_1        conda-forge     Cached\n",
            "  + conda                                            24.7.1  py312h7900ff3_0        conda-forge        1MB\n",
            "  - conda-libmamba-solver                            25.3.0  pyhd8ed1ab_0           conda-forge     Cached\n",
            "  + conda-libmamba-solver                            24.9.0  pyhd8ed1ab_0           conda-forge       42kB\n",
            "  - fmt                                              11.1.4  h07f6e7f_1             conda-forge     Cached\n",
            "  + fmt                                              10.2.1  h00ab1b0_0             conda-forge      194kB\n",
            "  - icu                                                75.1  he02047a_0             conda-forge     Cached\n",
            "  + icu                                                73.2  h59595ed_0             conda-forge       12MB\n",
            "  - libarchive                                        3.7.7  h75ea233_4             conda-forge     Cached\n",
            "  + libarchive                                        3.7.4  hfca40fe_0             conda-forge      872kB\n",
            "  - libcurl                                          8.14.1  h332b0f4_0             conda-forge     Cached\n",
            "  + libcurl                                           8.8.0  hca28451_1             conda-forge      410kB\n",
            "  - libmamba                                          2.1.1  h430c389_0             conda-forge     Cached\n",
            "  + libmamba                                          1.5.8  had39da4_0             conda-forge        2MB\n",
            "  - libmambapy                                        2.1.1  py312h07448e0_0        conda-forge     Cached\n",
            "  + libmambapy                                        1.5.8  py312hd9e9ff6_0        conda-forge      306kB\n",
            "  - libnghttp2                                       1.64.0  h161d5f1_0             conda-forge     Cached\n",
            "  + libnghttp2                                       1.58.0  h47da74e_1             conda-forge      632kB\n",
            "  - libsolv                                          0.7.34  h9463b59_0             conda-forge     Cached\n",
            "  + libsolv                                          0.7.29  ha6fb4c9_0             conda-forge      471kB\n",
            "  - libsqlite                                        3.50.3  hee844dc_1             conda-forge     Cached\n",
            "  + libsqlite                                        3.46.0  hde9e2c9_0             conda-forge      865kB\n",
            "  - libssh2                                          1.11.1  hcf80075_0             conda-forge     Cached\n",
            "  + libssh2                                          1.11.0  h0841786_0             conda-forge      271kB\n",
            "  - libxml2                                          2.13.8  h4bc477f_0             conda-forge     Cached\n",
            "  + libxml2                                          2.12.7  hc051c1a_1             conda-forge      705kB\n",
            "  - libzlib                                           1.3.1  hb9d3cd8_2             conda-forge     Cached\n",
            "  + libzlib                                          1.2.13  h4ab18f5_6             conda-forge       62kB\n",
            "  - lz4-c                                            1.10.0  h5888daf_1             conda-forge     Cached\n",
            "  + lz4-c                                             1.9.4  hcb278e6_0             conda-forge      143kB\n",
            "  - mamba                                             2.1.1  had4a41a_0             conda-forge     Cached\n",
            "  + mamba                                             1.5.8  py312h9460a1c_0        conda-forge       65kB\n",
            "  - python                                          3.12.11  h9e4cc4f_0_cpython     conda-forge     Cached\n",
            "  + python                                           3.12.3  hab00c5b_0_cpython     conda-forge       32MB\n",
            "  - zstd                                              1.5.7  hb8e6e7a_2             conda-forge     Cached\n",
            "  + zstd                                              1.5.6  ha6fb4c9_0             conda-forge      555kB\n",
            "\n",
            "  Summary:\n",
            "\n",
            "  Install: 271 packages\n",
            "  Change: 1 packages\n",
            "  Downgrade: 18 packages\n",
            "\n",
            "  Total download: 1GB\n",
            "\n",
            "────────────────────────────────────────────────────────────────────────────────────────────────────────────\n",
            "\n",
            "\n",
            "\n",
            "Transaction starting\n",
            "Unlinking icu-75.1-he02047a_0\n",
            "Unlinking libzlib-1.3.1-hb9d3cd8_2\n",
            "Unlinking fmt-11.1.4-h07f6e7f_1\n",
            "Unlinking lz4-c-1.10.0-h5888daf_1\n",
            "Unlinking libsolv-0.7.34-h9463b59_0\n",
            "Unlinking libssh2-1.11.1-hcf80075_0\n",
            "Unlinking libnghttp2-1.64.0-h161d5f1_0\n",
            "Unlinking zstd-1.5.7-hb8e6e7a_2\n",
            "Unlinking tk-8.6.13-noxft_hd72426e_102\n",
            "Unlinking libsqlite-3.50.3-hee844dc_1\n",
            "Unlinking libxml2-2.13.8-h4bc477f_0\n",
            "Unlinking libcurl-8.14.1-h332b0f4_0\n",
            "Unlinking python-3.12.11-h9e4cc4f_0_cpython\n",
            "Unlinking libarchive-3.7.7-h75ea233_4\n",
            "Unlinking libmamba-2.1.1-h430c389_0\n",
            "Unlinking libmambapy-2.1.1-py312h07448e0_0\n",
            "Unlinking conda-25.3.1-py312h7900ff3_1\n",
            "Unlinking conda-libmamba-solver-25.3.0-pyhd8ed1ab_0\n",
            "Unlinking mamba-2.1.1-had4a41a_0\n",
            "Linking font-ttf-dejavu-sans-mono-2.37-hab24e00_0\n",
            "Linking font-ttf-ubuntu-0.83-h77eed37_3\n",
            "Linking font-ttf-inconsolata-3.000-h77eed37_0\n",
            "Linking font-ttf-source-code-pro-2.038-h77eed37_0\n",
            "Linking libgcc-devel_linux-64-13.4.0-hba01cd7_107\n",
            "Linking kernel-headers_linux-64-5.14.0-he073ed8_2\n",
            "Linking fonts-conda-forge-1-0\n",
            "Linking sysroot_linux-64-2.34-h087de78_2\n",
            "Linking fonts-conda-ecosystem-1-0\n",
            "Linking liblzma-devel-5.8.1-hb9d3cd8_2\n",
            "Linking xz-gpl-tools-5.8.1-hbcc6ac9_2\n",
            "Linking xz-tools-5.8.1-hb9d3cd8_2\n",
            "Linking icu-73.2-h59595ed_0\n",
            "Linking libzlib-1.2.13-h4ab18f5_6\n",
            "Linking fmt-10.2.1-h00ab1b0_0\n",
            "Linking lz4-c-1.9.4-hcb278e6_0\n",
            "Linking giflib-5.2.2-hd590300_0\n",
            "Linking expat-2.7.1-hecca717_0\n",
            "Linking perl-5.32.1-7_hd590300_perl5\n",
            "Linking pbzip2-1.1.13-h1fcc475_2\n",
            "Linking isa-l-2.31.1-hb9d3cd8_1\n",
            "Linking zlib-ng-2.2.5-hde8ca8f_0\n",
            "Linking alsa-lib-1.2.11-hd590300_1\n",
            "Linking libjpeg-turbo-3.1.0-hb9d3cd8_0\n",
            "Linking pthread-stubs-0.4-hb9d3cd8_1002\n",
            "Linking xorg-libxau-1.0.12-hb9d3cd8_0\n",
            "Linking xorg-libxdmcp-1.1.5-hb9d3cd8_0\n",
            "Linking libdeflate-1.20-hd590300_0\n",
            "Linking graphite2-1.3.14-hecca717_2\n",
            "Linking pixman-0.46.4-h54a6638_1\n",
            "Linking xorg-libice-1.1.2-hb9d3cd8_0\n",
            "Linking mysql-connector-c-6.1.11-h659d440_1008\n",
            "Linking unzip-6.0-hb03c661_4\n",
            "Linking mysql-common-8.3.0-hf1915f5_4\n",
            "Linking xorg-inputproto-2.3.2-hb9d3cd8_1003\n",
            "Linking xorg-recordproto-1.14.2-hb9d3cd8_1003\n",
            "Linking xorg-kbproto-1.0.7-hb9d3cd8_1003\n",
            "Linking xorg-xproto-7.0.31-hb9d3cd8_1008\n",
            "Linking xorg-renderproto-0.11.1-hb9d3cd8_1003\n",
            "Linking xorg-xextproto-7.3.0-hb9d3cd8_1004\n",
            "Linking libntlm-1.8-hb9d3cd8_0\n",
            "Linking libevent-2.1.12-hf998b51_1\n",
            "Linking lerc-4.0.0-h0aef613_1\n",
            "Linking libwebp-base-1.6.0-hd42ef1d_0\n",
            "Linking libabseil-20240116.2-cxx17_he02047a_1\n",
            "Linking libsanitizer-13.4.0-h14bf0c3_7\n",
            "Linking pcre-8.45-h9c3ff4c_0\n",
            "Linking libgfortran5-15.1.0-hcea5267_5\n",
            "Linking argtable2-2.13-hd590300_1004\n",
            "Linking gmp-6.3.0-hac33072_2\n",
            "Linking mpi-1.0-openmpi\n",
            "Linking libtasn1-4.20.0-hb03c661_1\n",
            "Linking libdb-6.2.32-h9c3ff4c_0\n",
            "Linking libunistring-0.9.10-h7f98852_0\n",
            "Linking libasprintf-0.25.1-h3f43e3d_1\n",
            "Linking libgettextpo-0.25.1-h3f43e3d_1\n",
            "Linking binutils_impl_linux-64-2.44-h4bf12b8_1\n",
            "Linking xz-5.8.1-hbcc6ac9_2\n",
            "Linking pcre2-10.43-hcad00b1_0\n",
            "Linking pigz-2.8-h2797004_0\n",
            "Linking zlib-1.2.13-h4ab18f5_6\n",
            "Linking libpng-1.6.43-h2797004_0\n",
            "Linking libcups-2.3.3-h4637d8d_4\n",
            "Linking libsolv-0.7.29-ha6fb4c9_0\n",
            "Linking libssh2-1.11.0-h0841786_0\n",
            "Linking libnghttp2-1.58.0-h47da74e_1\n",
            "Linking zstd-1.5.6-ha6fb4c9_0\n",
            "Linking tk-8.6.13-noxft_h4845f30_101\n",
            "Linking libsqlite-3.46.0-hde9e2c9_0\n",
            "Linking perl-b-cow-0.007-pl5321hb9d3cd8_1\n",
            "Linking perl-xml-parser-2.44_01-pl5321hf2e2c51_1004\n",
            "Linking perl-compress-raw-zlib-2.214-pl5321h4dac143_0\n",
            "Linking perl-compress-raw-bzip2-2.214-pl5321hda65f42_0\n",
            "Linking perl-inc-latest-0.500-pl5321ha770c72_0\n",
            "Linking libxcb-1.15-h0b41bf4_0\n",
            "Linking xorg-libsm-1.2.6-he73a12e_0\n",
            "Linking xorg-fixesproto-5.0-hb9d3cd8_1003\n",
            "Linking cyrus-sasl-2.1.28-hd9c7081_0\n",
            "Linking libprotobuf-4.25.3-h08a7969_0\n",
            "Linking libgfortran-15.1.0-h69a702a_5\n",
            "Linking nettle-3.10.1-h4a9d5aa_0\n",
            "Linking mpfr-4.2.1-h90cbb55_3\n",
            "Linking p11-kit-0.24.1-hc5aa10d_0\n",
            "Linking perl-db_file-1.858-pl5321hb9d3cd8_0\n",
            "Linking libidn2-2.3.8-hfac485b_1\n",
            "Linking gcc_impl_linux-64-13.4.0-h69c5793_7\n",
            "Linking binutils_linux-64-2.44-h4852527_1\n",
            "Linking libxml2-2.12.7-hc051c1a_1\n",
            "Linking libglib-2.80.2-hf974151_0\n",
            "Linking freetype-2.12.1-h267a509_2\n",
            "Linking mysql-libs-8.3.0-hca2cd23_4\n",
            "Linking mysql-client-8.3.0-h545f5f4_4\n",
            "Linking libtiff-4.6.0-h1dd3fc0_3\n",
            "Linking libcurl-8.8.0-hca28451_1\n",
            "Linking python-3.12.3-hab00c5b_0_cpython\n",
            "Linking perl-module-build-0.4234-pl5321ha770c72_1\n",
            "Linking xorg-libx11-1.8.9-h8ee46fc_0\n",
            "Linking mysql-server-8.3.0-h27b9c54_4\n",
            "Linking libgfortran-ng-15.1.0-h69a702a_5\n",
            "Linking libopenblas-0.3.30-pthreads_h94d23a6_3\n",
            "Linking gawk-5.3.1-hcd3d067_0\n",
            "Linking gnutls-3.8.9-h86084c0_1\n",
            "Linking gcc_linux-64-13.4.0-h0a5b801_12\n",
            "Linking libarchive-3.7.4-hfca40fe_0\n",
            "Linking fontconfig-2.14.2-h14ed4e7_0\n",
            "Linking mysql-devel-8.3.0-hf1915f5_4\n",
            "Linking lcms2-2.16-hb7c19ff_0\n",
            "Linking python-zlib-ng-1.0.0-py312hd86e735_0\n",
            "Linking python-isal-1.8.0-py312h4c3975b_0\n",
            "Linking xorg-libxt-1.3.0-hd590300_1\n",
            "Linking xorg-libxfixes-5.0.3-h7f98852_1004\n",
            "Linking xorg-libxext-1.3.4-h0b41bf4_2\n",
            "Linking xorg-libxrender-0.9.11-hd590300_0\n",
            "Linking openmpi-4.1.6-hc5af2df_101\n",
            "\n",
            "Linking libblas-3.9.0-38_h4a7cf45_openblas\n",
            "Linking mafft-7.526-h4bc722e_0\n",
            "Linking libmamba-1.5.8-had39da4_0\n",
            "Linking mysql-8.3.0-h27aab58_4\n",
            "Linking xorg-libxi-1.7.10-h4bc722e_1\n",
            "Linking cairo-1.18.0-h3faef2a_0\n",
            "Linking libcblas-3.9.0-38_h0358290_openblas\n",
            "Linking libmambapy-1.5.8-py312hd9e9ff6_0\n",
            "Linking xorg-libxtst-1.2.5-h4bc722e_0\n",
            "Linking harfbuzz-9.0.0-hfac3d4d_0\n",
            "Linking gsl-2.7-he838d99_0\n",
            "Linking conda-24.7.1-py312h7900ff3_0\n",
            "Linking openjdk-17.0.11-h24d6bf4_0\n",
            "Linking clustalw-2.1-h9948957_12\n",
            "Linking paml-4.10.9-h7b50bb2_1\n",
            "Linking consan-1.2-h7b50bb2_7\n",
            "Linking dialign-tx-1.0.2-h7b50bb2_8\n",
            "Linking famsa-2.2.3-h9948957_1\n",
            "Linking kalign2-2.04-h7b50bb2_8\n",
            "Linking muscle-3.8.1551-h9948957_9\n",
            "Linking mustang-3.2.4-h9948957_0\n",
            "Linking prank-170427-h9948957_1\n",
            "Linking probconsrna-1.10-h9f5acd7_4\n",
            "Linking proda-1.0-h503566f_7\n",
            "Linking sap-1.1.3-h7b50bb2_5\n",
            "Linking fasttree-2.2.0-h7b50bb2_0\n",
            "Linking raxml-8.2.13-h7b50bb2_3\n",
            "Linking perl-version-0.9924-pl5321hec16e2b_2\n",
            "Linking perl-io-tty-1.16-pl5321hec16e2b_1\n",
            "Linking perl-sub-uplevel-0.2800-pl5321hec16e2b_4\n",
            "Linking perl-socket-2.027-pl5321h5c03b87_6\n",
            "Linking perl-url-encode-0.03-pl5321h9ee0642_1\n",
            "Linking perl-digest-md5-2.58-pl5321hec16e2b_1\n",
            "Linking perl-data-dump-1.25-pl5321h7b50bb2_2\n",
            "Linking perl-list-moreutils-xs-0.430-pl5321h7b50bb2_5\n",
            "Linking perl-mime-base64-3.16-pl5321hec16e2b_2\n",
            "Linking perl-data-dumper-2.183-pl5321hec16e2b_1\n",
            "Linking perl-scalar-list-utils-1.62-pl5321hec16e2b_1\n",
            "Linking perl-dbi-1.643-pl5321hec16e2b_1\n",
            "Linking bwa-0.7.18-he4a0461_1\n",
            "Linking clustalo-1.2.4-h503566f_10\n",
            "Linking bedtools-2.31.1-hf5e1c6e_2\n",
            "Linking probcons-1.12-h8b12597_1\n",
            "Linking perl-perlio-gzip-0.20-pl5321he4a0461_6\n",
            "Linking perl-clone-0.46-pl5321hec16e2b_0\n",
            "Linking tmalign-20170708-h8d36177_7\n",
            "Linking htslib-1.21-h5efdd21_0\n",
            "Linking blast-legacy-2.2.26-hf7ff83a_5\n",
            "Linking perl-devel-checklib-1.16-pl5321h7b50bb2_1\n",
            "Linking hmmer-3.4-hb6cb901_4\n",
            "Linking viennarna-2.7.0-py312pl5321h9f0f9fd_0\n",
            "Linking phylip-3.697-h7b50bb2_3\n",
            "Linking perl-dbd-sqlite-1.76-pl5321h7b50bb2_2\n",
            "Linking samtools-1.21-h50ea8bc_0\n",
            "Linking poa-2.0-h7b50bb2_6\n",
            "Linking perl-algorithm-diff-1.1903-pl5321hdfd78af_3\n",
            "Linking perl-mozilla-ca-20250602-pl5321hdfd78af_0\n",
            "Linking perl-devel-stacktrace-2.04-pl5321hdfd78af_1\n",
            "Linking perl-class-data-inheritable-0.09-pl5321hdfd78af_0\n",
            "Linking perl-test-deep-1.130-pl5321hdfd78af_0\n",
            "Linking perl-common-sense-3.75-pl5321hdfd78af_0\n",
            "Linking perl-html-tagset-3.24-pl5321hdfd78af_0\n",
            "Linking perl-test-needs-0.002009-pl5321hdfd78af_0\n",
            "Linking perl-io-html-1.004-pl5321hdfd78af_0\n",
            "Linking perl-try-tiny-0.31-pl5321hdfd78af_1\n",
            "Linking perl-timedate-2.33-pl5321hdfd78af_2\n",
            "Linking perl-getopt-long-2.58-pl5321hdfd78af_0\n",
            "Linking perl-base-2.23-pl5321hdfd78af_2\n",
            "Linking perl-file-sort-1.01-pl5321hdfd78af_3\n",
            "Linking perl-params-check-0.38-pl5321hdfd78af_2\n",
            "Linking perl-locale-maketext-simple-0.21-pl5321hdfd78af_3\n",
            "Linking perl-exporter-tiny-1.002002-pl5321hdfd78af_0\n",
            "Linking perl-digest-hmac-1.05-pl5321hdfd78af_0\n",
            "Linking perl-test-harness-3.44-pl5321hdfd78af_0\n",
            "Linking perl-module-load-0.34-pl5321hdfd78af_0\n",
            "Linking perl-file-spec-3.48_01-pl5321hdfd78af_2\n",
            "Linking perl-parent-0.236-pl5321hdfd78af_2\n",
            "Linking perl-exporter-5.72-pl5321hdfd78af_2\n",
            "Linking perl-constant-1.33-pl5321hdfd78af_2\n",
            "Linking perl-capture-tiny-0.48-pl5321hdfd78af_2\n",
            "Linking perl-extutils-manifest-1.75-pl5321hdfd78af_0\n",
            "Linking perl-xml-xpathengine-0.14-pl5321hdfd78af_3\n",
            "Linking perl-xml-regexp-0.04-pl5321hdfd78af_3\n",
            "Linking perl-io-string-1.08-pl5321hdfd78af_4\n",
            "Linking perl-libxml-perl-0.08-pl5321hdfd78af_3\n",
            "Linking perl-bioperl-core-1.7.8-pl5321hdfd78af_1\n",
            "Linking dendropy-5.0.8-pyhdfd78af_1\n",
            "Linking gatk4-4.6.2.0-py310hdfd78af_0\n",
            "Linking fastqc-0.12.1-hdfd78af_0\n",
            "Linking perl-module-corelist-5.20251022-pl5321hdfd78af_0\n",
            "Linking perl-ipc-run-20250809.0-pl5321hdfd78af_0\n",
            "Linking perl-test-warn-0.36-pl5321hdfd78af_2\n",
            "Linking perl-test-exception-0.43-pl5321hdfd78af_3\n",
            "Linking perl-test-requiresinternet-0.05-pl5321hdfd78af_1\n",
            "Linking perl-text-parsewords-3.31-pl5321hdfd78af_0\n",
            "Linking perl-text-diff-1.45-pl5321hdfd78af_1\n",
            "Linking perl-exception-class-1.45-pl5321hdfd78af_0\n",
            "Linking perl-types-serialiser-1.01-pl5321hdfd78af_0\n",
            "Linking perl-list-moreutils-0.430-pl5321hdfd78af_0\n",
            "Linking perl-ntlm-1.09-pl5321hdfd78af_5\n",
            "Linking perl-text-csv-2.01-pl5321hdfd78af_0\n",
            "Linking perl-perl-ostype-1.010-pl5321hdfd78af_2\n",
            "Linking perl-bio-searchio-hmmer-1.7.3-pl5321hdfd78af_0\n",
            "Linking perl-bio-tools-run-remoteblast-1.7.3-pl5321hdfd78af_0\n",
            "Linking perl-bio-db-swissprot-1.7.4-pl5321hdfd78af_0\n",
            "Linking perl-bio-db-refseq-1.7.4-pl5321hdfd78af_0\n",
            "Linking perl-bio-db-embl-1.7.4-pl5321hdfd78af_0\n",
            "Linking perl-test-differences-0.69-pl5321hdfd78af_0\n",
            "Linking perl-bio-procedural-1.7.4-pl5321hdfd78af_1\n",
            "Linking perl-test-most-0.38-pl5321hdfd78af_0\n",
            "Linking perl-bio-coordinate-1.007001-pl5321hdfd78af_3\n",
            "Linking perl-extutils-makemaker-7.70-pl5321hd8ed1ab_0\n",
            "Linking sniffio-1.3.1-pyhd8ed1ab_1\n",
            "Linking xopen-2.0.2-pyh707e725_2\n",
            "Linking conda-libmamba-solver-24.9.0-pyhd8ed1ab_0\n",
            "Linking perl-file-path-2.18-pl5321hd8ed1ab_0\n",
            "Linking dnspython-2.8.0-pyhcf101f3_0\n",
            "Linking perl-net-ssleay-1.92-pl5321hf14f497_1\n",
            "Linking mamba-1.5.8-py312h9460a1c_0\n",
            "Linking pymongo-4.15.3-py312h1289d80_0\n",
            "Linking perl-encode-3.19-pl5321hec16e2b_1\n",
            "Linking perl-extutils-cppguess-0.26-pl5321h9948957_4\n",
            "Linking perl-bio-samtools-1.43-pl5321he4a0461_4\n",
            "Linking perl-json-xs-4.04-pl5321h9948957_0\n",
            "Linking dnaio-1.2.2-py312hf67a6ed_0\n",
            "Linking pasta-1.9.2-py312hccd54bf_5\n",
            "Linking perl-io-compress-2.213-pl5321h503566f_0\n",
            "Linking perl-set-intervaltree-0.12-pl5321h503566f_6\n",
            "Linking cutadapt-5.2-py312h0fa9677_0\n",
            "Linking t-coffee-13.46.2.7c9e712d-pl5321hb2a3317_0\n",
            "Linking perl-carp-1.38-pl5321hdfd78af_4\n",
            "Linking perl-io-socket-ssl-2.074-pl5321hdfd78af_0\n",
            "Linking perl-encode-locale-1.05-pl5321hdfd78af_7\n",
            "Linking perl-bioperl-run-1.007003-pl5321hdfd78af_0\n",
            "Linking perl-json-4.10-pl5321hdfd78af_1\n",
            "Linking perl-module-metadata-1.000038-pl5321hdfd78af_0\n",
            "Linking perl-time-local-1.35-pl5321hdfd78af_0\n",
            "Linking perl-file-slurper-0.014-pl5321hdfd78af_0\n",
            "Linking perl-business-isbn-data-20210112.006-pl5321hdfd78af_0\n",
            "Linking perl-lwp-mediatypes-6.04-pl5321hdfd78af_1\n",
            "Linking perl-bio-asn1-entrezgene-1.73-pl5321hdfd78af_3\n",
            "Linking perl-bio-tools-run-alignment-tcoffee-1.7.4-pl5321hdfd78af_5\n",
            "Linking perl-bio-tools-run-alignment-clustalw-1.7.4-pl5321hdfd78af_3\n",
            "Linking perl-bio-tools-phylo-paml-1.7.3-pl5321hdfd78af_3\n",
            "Linking perl-module-load-conditional-0.68-pl5321hdfd78af_3\n",
            "Linking perl-http-date-6.06-pl5321hdfd78af_0\n",
            "Linking perl-tree-dag_node-1.35-pl5321hdfd78af_0\n",
            "Linking perl-business-isbn-3.007-pl5321hdfd78af_0\n",
            "Linking perl-file-listing-6.16-pl5321hdfd78af_0\n",
            "Linking perl-http-cookiejar-lwp-0.014-pl5321hdfd78af_0\n",
            "Linking perl-uri-5.12-pl5321hdfd78af_0\n",
            "Linking perl-net-http-6.24-pl5321hdfd78af_0\n",
            "Linking perl-www-robotrules-6.02-pl5321hdfd78af_4\n",
            "Linking perl-http-message-7.01-pl5321hdfd78af_0\n",
            "Linking perl-lwp-protocol-https-6.14-pl5321hdfd78af_1\n",
            "Linking perl-http-daemon-6.16-pl5321hdfd78af_0\n",
            "Linking perl-http-cookies-6.11-pl5321hdfd78af_0\n",
            "Linking perl-http-negotiate-6.01-pl5321hdfd78af_4\n",
            "Linking perl-file-temp-0.2304-pl5321hd8ed1ab_0\n",
            "Linking perl-time-hires-1.9764-pl5321h7b50bb2_6\n",
            "Linking perl-html-parser-3.81-pl5321h4ac6f70_1\n",
            "Linking perl-dbd-mysql-4.050-pl5321h9948957_3\n",
            "Linking perl-ipc-cmd-1.04-pl5321hdfd78af_1\n",
            "Linking perl-libwww-perl-6.81-pl5321hdfd78af_0\n",
            "Linking perl-extutils-cbuilder-0.280230-pl5321hdfd78af_2\n",
            "Linking perl-xml-dom-1.46-pl5321hdfd78af_1\n",
            "Linking perl-extutils-parsexs-3.60-pl5321hdfd78af_0\n",
            "Linking perl-xml-dom-xpath-0.14-pl5321hdfd78af_2\n",
            "Linking perl-bio-featureio-1.6.905-pl5321hdfd78af_4\n",
            "Linking perl-bioperl-1.7.8-hdfd78af_1\n",
            "Linking perl-sereal-decoder-5.004-pl5321h7b50bb2_0\n",
            "Linking perl-bio-db-hts-3.01-pl5321h577a1d6_12\n",
            "Linking perl-bio-bigfile-1.07-pl5321h41f7678_5\n",
            "Linking perl-sereal-encoder-5.004-pl5321h7b50bb2_0\n",
            "Linking perl-sereal-5.004-pl5321hdfd78af_0\n",
            "Linking ensembl-vep-115.2-pl5321h2a3209d_1\n",
            "\n",
            "\n",
            "Transaction finished\n",
            "\n"
          ]
        },
        {
          "output_type": "stream",
          "name": "stderr",
          "text": [
            "warning  libmamba [mysql-libs-8.3.0-hca2cd23_4] The following files were already present in the environment:\n",
            "    - lib/libmysqlclient.so\n",
            "warning  libmamba [mysql-server-8.3.0-h27b9c54_4] The following files were already present in the environment:\n",
            "    - bin/my_print_defaults\n",
            "    - bin/perror\n",
            "warning  libmamba [mysql-devel-8.3.0-hf1915f5_4] The following files were already present in the environment:\n",
            "    - bin/mysql_config\n",
            "    - lib/libmysqlclient.a\n",
            "/tmp/mambaf8pdc40xm1e: line 2: /usr/local/bin/mamba (deleted): No such file or directory\n",
            "/tmp/mambaf8pdc40xm1e: line 3: syntax error near unexpected token `deleted'\n",
            "/tmp/mambaf8pdc40xm1e: line 3: `mamba (deleted) activate \"/usr/local\"'\n",
            "/tmp/mambafnmqw4zzv4h: line 2: /usr/local/bin/mamba (deleted): No such file or directory\n",
            "/tmp/mambafnmqw4zzv4h: line 3: syntax error near unexpected token `deleted'\n",
            "/tmp/mambafnmqw4zzv4h: line 3: `mamba (deleted) activate \"/usr/local\"'\n"
          ]
        }
      ],
      "source": [
        "%%bash\n",
        "# Instalar os software que vamos usar neste pipeline\n",
        "mamba install -c bioconda -c conda-forge fastqc bwa gatk4 cutadapt samtools bedtools ensembl-vep=115 -y"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": null,
      "metadata": {
        "id": "XuQiIitLbJ5m",
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "outputId": "11883c19-7ca3-4025-ba4e-5dc026bd7b78",
        "collapsed": true
      },
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "#----------------------------------#\n",
            "# ENSEMBL VARIANT EFFECT PREDICTOR #\n",
            "#----------------------------------#\n",
            "\n",
            "Versions:\n",
            "  ensembl              : 115.266b84d\n",
            "  ensembl-compara      : 115.ae48a7a\n",
            "  ensembl-funcgen      : 115.57f7061\n",
            "  ensembl-io           : 115.25061d3\n",
            "  ensembl-variation    : 115.b7c2637\n",
            "  ensembl-vep          : 115.2\n",
            "\n",
            "Help: dev@ensembl.org , helpdesk@ensembl.org\n",
            "Twitter: @ensembl\n",
            "\n",
            "http://www.ensembl.org/info/docs/tools/vep/script/index.html\n",
            "\n",
            "Usage:\n",
            "./vep [--cache|--offline|--database] [arguments]\n",
            "\n",
            "Basic options\n",
            "=============\n",
            "\n",
            "--help                 Display this message and quit\n",
            "\n",
            "-i | --input_file      Input file\n",
            "-o | --output_file     Output file\n",
            "--force_overwrite      Force overwriting of output file\n",
            "--species [species]    Species to use [default: \"human\"]\n",
            "\n",
            "--everything           Shortcut switch to turn on commonly used options. See web\n",
            "                       documentation for details [default: off]\n",
            "--fork [num_forks]     Use forking to improve script runtime\n",
            "\n",
            "For full option documentation see:\n",
            "http://www.ensembl.org/info/docs/tools/vep/script/vep_options.html\n",
            "\n"
          ]
        }
      ],
      "source": [
        "%%bash\n",
        "# Testa se as bibliotecas foram instaladas corretamente\n",
        "vep --help"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": null,
      "metadata": {
        "id": "t9MWHnkWgwWz"
      },
      "outputs": [],
      "source": [
        "%%bash\n",
        "# Criar as pastas para organizar os arquivos que serão gerados\n",
        "mkdir -p raw fastqc cutadapt bwa gatk vep reference"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": null,
      "metadata": {
        "id": "vyehK-nWQnQB"
      },
      "outputs": [],
      "source": [
        "%%bash\n",
        "# Mover arquivos para as suas respectivas pastas e descompactar fasta.gz\n",
        "mv *.fastq.gz raw/\n",
        "gunzip -c Chr12-reference.fasta.gz > reference/Chr12-reference.fasta\n",
        "rm -rf Chr12-reference.fasta.gz sample_data/"
      ]
    },
    {
      "cell_type": "markdown",
      "metadata": {
        "id": "qZ1G7s0DQcgi"
      },
      "source": [
        "**Checkpoint 1:**  \n",
        "Antes de seguir, verifique se as pastas foram criadas. Os arquivos fastqs do paciente deve estar dentro da pasta raw/ (2 arquivos) e o arquivo fasta da referência deve estar dentro da pasta reference/."
      ]
    },
    {
      "cell_type": "markdown",
      "metadata": {
        "id": "HbFjonTRGvxV"
      },
      "source": [
        "# Etapa 1 - Controle de Qualidade\n",
        "Filtrando os pares de bases que tiveram um score de qualidade baixos."
      ]
    },
    {
      "cell_type": "code",
      "execution_count": null,
      "metadata": {
        "id": "bwZ6imuMZesX",
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "outputId": "590da40f-b2f6-45ed-86b9-79203c27d7a7",
        "collapsed": true
      },
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "application/gzip\n",
            "application/gzip\n",
            "Analysis complete for caso2-sindrome-metabolica_R1.fastq.gz\n",
            "Analysis complete for caso2-sindrome-metabolica_R2.fastq.gz\n"
          ]
        },
        {
          "output_type": "stream",
          "name": "stderr",
          "text": [
            "Started analysis of caso2-sindrome-metabolica_R1.fastq.gz\n",
            "Approx 5% complete for caso2-sindrome-metabolica_R1.fastq.gz\n",
            "Approx 10% complete for caso2-sindrome-metabolica_R1.fastq.gz\n",
            "Approx 15% complete for caso2-sindrome-metabolica_R1.fastq.gz\n",
            "Approx 20% complete for caso2-sindrome-metabolica_R1.fastq.gz\n",
            "Approx 25% complete for caso2-sindrome-metabolica_R1.fastq.gz\n",
            "Approx 30% complete for caso2-sindrome-metabolica_R1.fastq.gz\n",
            "Approx 35% complete for caso2-sindrome-metabolica_R1.fastq.gz\n",
            "Approx 40% complete for caso2-sindrome-metabolica_R1.fastq.gz\n",
            "Approx 45% complete for caso2-sindrome-metabolica_R1.fastq.gz\n",
            "Approx 50% complete for caso2-sindrome-metabolica_R1.fastq.gz\n",
            "Approx 55% complete for caso2-sindrome-metabolica_R1.fastq.gz\n",
            "Approx 60% complete for caso2-sindrome-metabolica_R1.fastq.gz\n",
            "Approx 65% complete for caso2-sindrome-metabolica_R1.fastq.gz\n",
            "Approx 70% complete for caso2-sindrome-metabolica_R1.fastq.gz\n",
            "Approx 75% complete for caso2-sindrome-metabolica_R1.fastq.gz\n",
            "Approx 80% complete for caso2-sindrome-metabolica_R1.fastq.gz\n",
            "Approx 85% complete for caso2-sindrome-metabolica_R1.fastq.gz\n",
            "Approx 90% complete for caso2-sindrome-metabolica_R1.fastq.gz\n",
            "Approx 95% complete for caso2-sindrome-metabolica_R1.fastq.gz\n",
            "Approx 100% complete for caso2-sindrome-metabolica_R1.fastq.gz\n",
            "Started analysis of caso2-sindrome-metabolica_R2.fastq.gz\n",
            "Approx 5% complete for caso2-sindrome-metabolica_R2.fastq.gz\n",
            "Approx 10% complete for caso2-sindrome-metabolica_R2.fastq.gz\n",
            "Approx 15% complete for caso2-sindrome-metabolica_R2.fastq.gz\n",
            "Approx 20% complete for caso2-sindrome-metabolica_R2.fastq.gz\n",
            "Approx 25% complete for caso2-sindrome-metabolica_R2.fastq.gz\n",
            "Approx 30% complete for caso2-sindrome-metabolica_R2.fastq.gz\n",
            "Approx 35% complete for caso2-sindrome-metabolica_R2.fastq.gz\n",
            "Approx 40% complete for caso2-sindrome-metabolica_R2.fastq.gz\n",
            "Approx 45% complete for caso2-sindrome-metabolica_R2.fastq.gz\n",
            "Approx 50% complete for caso2-sindrome-metabolica_R2.fastq.gz\n",
            "Approx 55% complete for caso2-sindrome-metabolica_R2.fastq.gz\n",
            "Approx 60% complete for caso2-sindrome-metabolica_R2.fastq.gz\n",
            "Approx 65% complete for caso2-sindrome-metabolica_R2.fastq.gz\n",
            "Approx 70% complete for caso2-sindrome-metabolica_R2.fastq.gz\n",
            "Approx 75% complete for caso2-sindrome-metabolica_R2.fastq.gz\n",
            "Approx 80% complete for caso2-sindrome-metabolica_R2.fastq.gz\n",
            "Approx 85% complete for caso2-sindrome-metabolica_R2.fastq.gz\n",
            "Approx 90% complete for caso2-sindrome-metabolica_R2.fastq.gz\n",
            "Approx 95% complete for caso2-sindrome-metabolica_R2.fastq.gz\n",
            "Approx 100% complete for caso2-sindrome-metabolica_R2.fastq.gz\n"
          ]
        }
      ],
      "source": [
        "%%bash\n",
        "# Gerar o relatório da qualidade dos reads de sequenciamento\n",
        "fastqc -o fastqc/ raw/caso2-sindrome-metabolica_R1.fastq.gz raw/caso2-sindrome-metabolica_R2.fastq.gz"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": null,
      "metadata": {
        "id": "t_JPX-mNF0AS",
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "outputId": "d4563ed7-eb1c-4244-d7c3-b0dd253791b1",
        "collapsed": true
      },
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "This is cutadapt 5.2 with Python 3.12.3\n",
            "Command line parameters: -u 5 -U 5 -u -5 -U -5 -m 90 -o cutadapt/caso2-sindrome-metabolica-trimmed_R1.fastq.gz -p cutadapt/caso2-sindrome-metabolica-trimmed_R2.fastq.gz raw/caso2-sindrome-metabolica_R1.fastq.gz raw/caso2-sindrome-metabolica_R2.fastq.gz\n",
            "Processing paired-end reads on 1 core ...\n",
            "\n",
            "=== Summary ===\n",
            "\n",
            "Total read pairs processed:          3,000,000\n",
            "\n",
            "== Read fate breakdown ==\n",
            "Pairs that were too short:                   2 (0.0%)\n",
            "Pairs written (passing filters):     2,999,998 (100.0%)\n",
            "\n",
            "Total basepairs processed:   900,007,113 bp\n",
            "  Read 1:   450,024,803 bp\n",
            "  Read 2:   449,982,310 bp\n",
            "Total written (filtered):    840,006,660 bp (93.3%)\n",
            "  Read 1:   420,024,574 bp\n",
            "  Read 2:   419,982,086 bp\n"
          ]
        }
      ],
      "source": [
        "%%bash\n",
        "# Filtrar sequências muito pequenas e cortar as pontas\n",
        "cutadapt \\\n",
        "  -u 5 -U 5 \\\n",
        "  -u -5 -U -5 \\\n",
        "  -m 90 \\\n",
        "  -o cutadapt/caso2-sindrome-metabolica-trimmed_R1.fastq.gz \\\n",
        "  -p cutadapt/caso2-sindrome-metabolica-trimmed_R2.fastq.gz \\\n",
        "  raw/caso2-sindrome-metabolica_R1.fastq.gz \\\n",
        "  raw/caso2-sindrome-metabolica_R2.fastq.gz"
      ]
    },
    {
      "cell_type": "markdown",
      "source": [
        "**Chekpoint 1:**\\\n",
        "Foi possível observar que mantivemos quase todo o genoma. Quality score bom em grande parte do genoma."
      ],
      "metadata": {
        "id": "Z95WpepUMib1"
      }
    },
    {
      "cell_type": "markdown",
      "metadata": {
        "id": "JU46HSfKCDvs"
      },
      "source": [
        "# Etapa 2 - Mapeamento / Alinhamento\n",
        "Alinhamento do genoma do paciente com o genoma humano de referência gerando um arquivo .bam e .bai para análise no IGV."
      ]
    },
    {
      "cell_type": "code",
      "execution_count": null,
      "metadata": {
        "id": "MOMvG9EzCgkW",
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "outputId": "a712ac7f-1b0d-4016-b91f-f910a19ef7b1",
        "collapsed": true
      },
      "outputs": [
        {
          "output_type": "stream",
          "name": "stderr",
          "text": [
            "[bwa_index] Pack FASTA... 1.17 sec\n",
            "[bwa_index] Construct BWT for the packed sequence...\n",
            "[BWTIncCreate] textLength=266550618, availableWord=30755168\n",
            "[BWTIncConstructFromPacked] 10 iterations done. 50732074 characters processed.\n",
            "[BWTIncConstructFromPacked] 20 iterations done. 93723498 characters processed.\n",
            "[BWTIncConstructFromPacked] 30 iterations done. 131930362 characters processed.\n",
            "[BWTIncConstructFromPacked] 40 iterations done. 165884698 characters processed.\n",
            "[BWTIncConstructFromPacked] 50 iterations done. 196059354 characters processed.\n",
            "[BWTIncConstructFromPacked] 60 iterations done. 222874650 characters processed.\n",
            "[BWTIncConstructFromPacked] 70 iterations done. 246704106 characters processed.\n",
            "[BWTIncConstructFromPacked] 80 iterations done. 266550618 characters processed.\n",
            "[bwt_gen] Finished constructing BWT in 80 iterations.\n",
            "[bwa_index] 103.40 seconds elapse.\n",
            "[bwa_index] Update BWT... 0.99 sec\n",
            "[bwa_index] Pack forward-only FASTA... 0.71 sec\n",
            "[bwa_index] Construct SA from BWT and Occ... 45.53 sec\n",
            "[main] Version: 0.7.18-r1243-dirty\n",
            "[main] CMD: bwa index reference/Chr12-reference.fasta\n",
            "[main] Real time: 154.209 sec; CPU: 151.802 sec\n"
          ]
        }
      ],
      "source": [
        "%%bash\n",
        "# Indexar o genoma humano de referência para realizar o mapeamento e posterior remoção de contaminante\n",
        "samtools faidx reference/Chr12-reference.fasta\n",
        "bwa index reference/Chr12-reference.fasta"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": null,
      "metadata": {
        "id": "hvDSuj7PEK6c",
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "outputId": "7d6c65cb-50e3-4c0f-d5cf-7afe2413f50a",
        "collapsed": true
      },
      "outputs": [
        {
          "output_type": "stream",
          "name": "stderr",
          "text": [
            "[M::bwa_idx_load_from_disk] read 0 ALT contigs\n",
            "[M::process] read 71422 sequences (10000189 bp)...\n",
            "[M::process] read 71440 sequences (10000217 bp)...\n",
            "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 34793, 0, 0)\n",
            "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (319, 339, 359)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (239, 439)\n",
            "[M::mem_pestat] mean and std.dev: (340.01, 28.20)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (199, 479)\n",
            "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
            "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
            "[M::mem_process_seqs] Processed 71422 reads in 22.002 CPU sec, 21.847 real sec\n",
            "[M::process] read 71420 sequences (10000023 bp)...\n",
            "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 34716, 0, 0)\n",
            "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (320, 339, 359)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (242, 437)\n",
            "[M::mem_pestat] mean and std.dev: (340.12, 28.13)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (203, 476)\n",
            "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
            "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
            "[M::mem_process_seqs] Processed 71440 reads in 21.507 CPU sec, 21.277 real sec\n",
            "[M::process] read 71434 sequences (10000161 bp)...\n",
            "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 34749, 0, 0)\n",
            "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (319, 339, 359)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (239, 439)\n",
            "[M::mem_pestat] mean and std.dev: (339.72, 28.29)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (199, 479)\n",
            "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
            "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
            "[M::mem_process_seqs] Processed 71420 reads in 22.057 CPU sec, 21.937 real sec\n",
            "[M::process] read 71458 sequences (10000061 bp)...\n",
            "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 34722, 0, 0)\n",
            "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (319, 339, 359)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (239, 439)\n",
            "[M::mem_pestat] mean and std.dev: (340.06, 28.19)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (199, 479)\n",
            "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
            "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
            "[M::mem_process_seqs] Processed 71434 reads in 22.725 CPU sec, 22.552 real sec\n",
            "[M::process] read 71410 sequences (10000066 bp)...\n",
            "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 34737, 0, 0)\n",
            "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (319, 339, 359)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (239, 439)\n",
            "[M::mem_pestat] mean and std.dev: (339.92, 28.42)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (199, 479)\n",
            "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
            "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
            "[M::mem_process_seqs] Processed 71458 reads in 22.803 CPU sec, 22.607 real sec\n",
            "[M::process] read 71432 sequences (10000060 bp)...\n",
            "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 34709, 0, 0)\n",
            "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (319, 339, 359)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (239, 439)\n",
            "[M::mem_pestat] mean and std.dev: (339.95, 28.22)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (199, 479)\n",
            "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
            "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
            "[M::mem_process_seqs] Processed 71410 reads in 21.754 CPU sec, 21.798 real sec\n",
            "[M::process] read 71406 sequences (10000180 bp)...\n",
            "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 34758, 0, 0)\n",
            "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (319, 339, 359)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (239, 439)\n",
            "[M::mem_pestat] mean and std.dev: (340.12, 28.18)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (199, 479)\n",
            "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
            "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
            "[M::mem_process_seqs] Processed 71432 reads in 21.757 CPU sec, 21.599 real sec\n",
            "[M::process] read 71428 sequences (10000008 bp)...\n",
            "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 34743, 0, 0)\n",
            "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (319, 339, 360)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (237, 442)\n",
            "[M::mem_pestat] mean and std.dev: (340.14, 28.35)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (196, 483)\n",
            "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
            "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
            "[M::mem_process_seqs] Processed 71406 reads in 22.198 CPU sec, 22.005 real sec\n",
            "[M::process] read 71424 sequences (10000062 bp)...\n",
            "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 34792, 0, 0)\n",
            "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (319, 339, 359)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (239, 439)\n",
            "[M::mem_pestat] mean and std.dev: (339.76, 28.28)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (199, 479)\n",
            "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
            "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
            "[M::mem_process_seqs] Processed 71428 reads in 21.091 CPU sec, 20.840 real sec\n",
            "[M::process] read 71404 sequences (10000097 bp)...\n",
            "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 34710, 0, 0)\n",
            "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (319, 339, 359)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (239, 439)\n",
            "[M::mem_pestat] mean and std.dev: (339.81, 28.24)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (199, 479)\n",
            "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
            "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
            "[M::mem_process_seqs] Processed 71424 reads in 22.714 CPU sec, 22.528 real sec\n",
            "[M::process] read 71426 sequences (10000119 bp)...\n",
            "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 34719, 0, 0)\n",
            "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (319, 339, 359)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (239, 439)\n",
            "[M::mem_pestat] mean and std.dev: (340.01, 28.20)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (199, 479)\n",
            "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
            "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
            "[M::mem_process_seqs] Processed 71404 reads in 22.293 CPU sec, 22.153 real sec\n",
            "[M::process] read 71442 sequences (10000251 bp)...\n",
            "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 34721, 0, 0)\n",
            "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (319, 339, 359)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (239, 439)\n",
            "[M::mem_pestat] mean and std.dev: (339.92, 28.19)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (199, 479)\n",
            "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
            "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
            "[M::mem_process_seqs] Processed 71426 reads in 22.166 CPU sec, 21.955 real sec\n",
            "[M::process] read 71406 sequences (10000003 bp)...\n",
            "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 34766, 0, 0)\n",
            "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (320, 339, 359)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (242, 437)\n",
            "[M::mem_pestat] mean and std.dev: (339.93, 28.18)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (203, 476)\n",
            "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
            "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
            "[M::mem_process_seqs] Processed 71442 reads in 21.862 CPU sec, 21.905 real sec\n",
            "[M::process] read 71442 sequences (10000199 bp)...\n",
            "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 34732, 0, 0)\n",
            "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (319, 339, 360)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (237, 442)\n",
            "[M::mem_pestat] mean and std.dev: (340.28, 28.51)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (196, 483)\n",
            "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
            "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
            "[M::mem_process_seqs] Processed 71406 reads in 21.632 CPU sec, 21.460 real sec\n",
            "[M::process] read 71412 sequences (10000159 bp)...\n",
            "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 34768, 0, 0)\n",
            "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (319, 339, 359)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (239, 439)\n",
            "[M::mem_pestat] mean and std.dev: (339.90, 28.28)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (199, 479)\n",
            "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
            "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
            "[M::mem_process_seqs] Processed 71442 reads in 21.515 CPU sec, 21.296 real sec\n",
            "[M::process] read 71408 sequences (10000266 bp)...\n",
            "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 34721, 0, 0)\n",
            "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (319, 339, 359)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (239, 439)\n",
            "[M::mem_pestat] mean and std.dev: (339.47, 28.20)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (199, 479)\n",
            "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
            "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
            "[M::mem_process_seqs] Processed 71412 reads in 20.922 CPU sec, 20.886 real sec\n",
            "[M::process] read 71436 sequences (10000012 bp)...\n",
            "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 34759, 0, 0)\n",
            "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (319, 339, 359)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (239, 439)\n",
            "[M::mem_pestat] mean and std.dev: (339.75, 28.39)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (199, 479)\n",
            "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
            "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
            "[M::mem_process_seqs] Processed 71408 reads in 21.510 CPU sec, 21.349 real sec\n",
            "[M::process] read 71416 sequences (10000223 bp)...\n",
            "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 34720, 0, 0)\n",
            "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (319, 339, 359)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (239, 439)\n",
            "[M::mem_pestat] mean and std.dev: (339.89, 28.14)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (199, 479)\n",
            "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
            "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
            "[M::mem_process_seqs] Processed 71436 reads in 21.640 CPU sec, 21.471 real sec\n",
            "[M::process] read 71428 sequences (10000242 bp)...\n",
            "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 34719, 0, 0)\n",
            "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (319, 339, 359)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (239, 439)\n",
            "[M::mem_pestat] mean and std.dev: (339.92, 28.24)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (199, 479)\n",
            "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
            "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
            "[M::mem_process_seqs] Processed 71416 reads in 20.768 CPU sec, 20.820 real sec\n",
            "[M::process] read 71454 sequences (10000012 bp)...\n",
            "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 34722, 0, 0)\n",
            "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (319, 339, 359)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (239, 439)\n",
            "[M::mem_pestat] mean and std.dev: (340.06, 28.27)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (199, 479)\n",
            "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
            "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
            "[M::mem_process_seqs] Processed 71428 reads in 21.528 CPU sec, 21.314 real sec\n",
            "[M::process] read 71418 sequences (10000260 bp)...\n",
            "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 34740, 0, 0)\n",
            "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (319, 339, 359)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (239, 439)\n",
            "[M::mem_pestat] mean and std.dev: (340.01, 28.28)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (199, 479)\n",
            "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
            "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
            "[M::mem_process_seqs] Processed 71454 reads in 21.638 CPU sec, 21.438 real sec\n",
            "[M::process] read 71430 sequences (10000091 bp)...\n",
            "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 34701, 0, 0)\n",
            "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (319, 339, 359)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (239, 439)\n",
            "[M::mem_pestat] mean and std.dev: (340.03, 28.38)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (199, 479)\n",
            "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
            "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
            "[M::mem_process_seqs] Processed 71418 reads in 21.174 CPU sec, 21.149 real sec\n",
            "[M::process] read 71430 sequences (10000055 bp)...\n",
            "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 34748, 0, 0)\n",
            "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (319, 339, 359)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (239, 439)\n",
            "[M::mem_pestat] mean and std.dev: (339.66, 28.21)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (199, 479)\n",
            "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
            "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
            "[M::mem_process_seqs] Processed 71430 reads in 21.561 CPU sec, 21.417 real sec\n",
            "[M::process] read 71438 sequences (10000091 bp)...\n",
            "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 34743, 0, 0)\n",
            "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (319, 339, 359)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (239, 439)\n",
            "[M::mem_pestat] mean and std.dev: (339.97, 28.41)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (199, 479)\n",
            "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
            "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
            "[M::mem_process_seqs] Processed 71430 reads in 21.728 CPU sec, 21.519 real sec\n",
            "[M::process] read 71416 sequences (10000236 bp)...\n",
            "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 34711, 0, 0)\n",
            "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (319, 339, 359)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (239, 439)\n",
            "[M::mem_pestat] mean and std.dev: (339.99, 28.35)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (199, 479)\n",
            "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
            "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
            "[M::mem_process_seqs] Processed 71438 reads in 21.706 CPU sec, 21.694 real sec\n",
            "[M::process] read 71422 sequences (10000179 bp)...\n",
            "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 34778, 0, 0)\n",
            "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (319, 339, 359)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (239, 439)\n",
            "[M::mem_pestat] mean and std.dev: (339.66, 28.13)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (199, 479)\n",
            "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
            "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
            "[M::mem_process_seqs] Processed 71416 reads in 21.925 CPU sec, 21.770 real sec\n",
            "[M::process] read 71468 sequences (10000090 bp)...\n",
            "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 34724, 0, 0)\n",
            "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (319, 339, 359)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (239, 439)\n",
            "[M::mem_pestat] mean and std.dev: (339.85, 28.10)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (199, 479)\n",
            "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
            "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
            "[M::mem_process_seqs] Processed 71422 reads in 21.916 CPU sec, 21.730 real sec\n",
            "[M::process] read 71438 sequences (10000005 bp)...\n",
            "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 34735, 0, 0)\n",
            "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (319, 339, 359)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (239, 439)\n",
            "[M::mem_pestat] mean and std.dev: (339.72, 28.31)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (199, 479)\n",
            "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
            "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
            "[M::mem_process_seqs] Processed 71468 reads in 22.884 CPU sec, 22.727 real sec\n",
            "[M::process] read 71434 sequences (10000169 bp)...\n",
            "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 34776, 0, 0)\n",
            "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (319, 339, 359)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (239, 439)\n",
            "[M::mem_pestat] mean and std.dev: (339.87, 28.30)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (199, 479)\n",
            "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
            "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
            "[M::mem_process_seqs] Processed 71438 reads in 21.623 CPU sec, 21.651 real sec\n",
            "[M::process] read 71426 sequences (10000065 bp)...\n",
            "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 34740, 0, 0)\n",
            "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (319, 339, 359)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (239, 439)\n",
            "[M::mem_pestat] mean and std.dev: (340.06, 28.21)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (199, 479)\n",
            "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
            "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
            "[M::mem_process_seqs] Processed 71434 reads in 21.535 CPU sec, 21.406 real sec\n",
            "[M::process] read 71444 sequences (10000231 bp)...\n",
            "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 34750, 0, 0)\n",
            "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (319, 339, 359)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (239, 439)\n",
            "[M::mem_pestat] mean and std.dev: (340.18, 28.35)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (199, 479)\n",
            "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
            "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
            "[M::mem_process_seqs] Processed 71426 reads in 20.798 CPU sec, 20.619 real sec\n",
            "[M::process] read 71424 sequences (10000085 bp)...\n",
            "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 34727, 0, 0)\n",
            "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (319, 339, 360)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (237, 442)\n",
            "[M::mem_pestat] mean and std.dev: (340.06, 28.41)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (196, 483)\n",
            "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
            "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
            "[M::mem_process_seqs] Processed 71444 reads in 21.777 CPU sec, 21.618 real sec\n",
            "[M::process] read 71400 sequences (10000006 bp)...\n",
            "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 34686, 0, 0)\n",
            "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (319, 339, 359)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (239, 439)\n",
            "[M::mem_pestat] mean and std.dev: (339.96, 28.36)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (199, 479)\n",
            "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
            "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
            "[M::mem_process_seqs] Processed 71424 reads in 24.427 CPU sec, 26.206 real sec\n",
            "[M::process] read 71420 sequences (10000057 bp)...\n",
            "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 34701, 0, 0)\n",
            "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (319, 339, 359)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (239, 439)\n",
            "[M::mem_pestat] mean and std.dev: (339.91, 28.15)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (199, 479)\n",
            "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
            "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
            "[M::mem_process_seqs] Processed 71400 reads in 21.699 CPU sec, 21.490 real sec\n",
            "[M::process] read 71462 sequences (10000063 bp)...\n",
            "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 34731, 0, 0)\n",
            "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (319, 339, 360)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (237, 442)\n",
            "[M::mem_pestat] mean and std.dev: (339.98, 28.40)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (196, 483)\n",
            "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
            "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
            "[M::mem_process_seqs] Processed 71420 reads in 20.638 CPU sec, 20.432 real sec\n",
            "[M::process] read 71448 sequences (10000009 bp)...\n",
            "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 34716, 0, 0)\n",
            "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (319, 339, 359)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (239, 439)\n",
            "[M::mem_pestat] mean and std.dev: (340.00, 28.59)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (199, 479)\n",
            "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
            "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
            "[M::mem_process_seqs] Processed 71462 reads in 22.018 CPU sec, 21.823 real sec\n",
            "[M::process] read 71440 sequences (10000263 bp)...\n",
            "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 34758, 0, 0)\n",
            "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (319, 339, 359)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (239, 439)\n",
            "[M::mem_pestat] mean and std.dev: (340.04, 28.25)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (199, 479)\n",
            "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
            "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
            "[M::mem_process_seqs] Processed 71448 reads in 21.699 CPU sec, 21.511 real sec\n",
            "[M::process] read 71406 sequences (10000050 bp)...\n",
            "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 34679, 0, 0)\n",
            "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (319, 339, 359)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (239, 439)\n",
            "[M::mem_pestat] mean and std.dev: (339.89, 28.28)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (199, 479)\n",
            "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
            "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
            "[M::mem_process_seqs] Processed 71440 reads in 20.755 CPU sec, 20.528 real sec\n",
            "[M::process] read 71438 sequences (10000263 bp)...\n",
            "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 34691, 0, 0)\n",
            "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (319, 339, 359)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (239, 439)\n",
            "[M::mem_pestat] mean and std.dev: (339.83, 28.34)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (199, 479)\n",
            "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
            "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
            "[M::mem_process_seqs] Processed 71406 reads in 21.379 CPU sec, 21.167 real sec\n",
            "[M::process] read 71426 sequences (10000071 bp)...\n",
            "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 34702, 0, 0)\n",
            "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (319, 339, 359)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (239, 439)\n",
            "[M::mem_pestat] mean and std.dev: (339.91, 28.46)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (199, 479)\n",
            "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
            "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
            "[M::mem_process_seqs] Processed 71438 reads in 22.161 CPU sec, 22.006 real sec\n",
            "[M::process] read 71434 sequences (10000215 bp)...\n",
            "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 34781, 0, 0)\n",
            "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (319, 339, 359)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (239, 439)\n",
            "[M::mem_pestat] mean and std.dev: (339.88, 28.22)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (199, 479)\n",
            "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
            "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
            "[M::mem_process_seqs] Processed 71426 reads in 21.334 CPU sec, 21.077 real sec\n",
            "[M::process] read 71454 sequences (10000170 bp)...\n",
            "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 34750, 0, 0)\n",
            "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (319, 339, 359)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (239, 439)\n",
            "[M::mem_pestat] mean and std.dev: (339.55, 28.29)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (199, 479)\n",
            "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
            "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
            "[M::mem_process_seqs] Processed 71434 reads in 21.726 CPU sec, 21.564 real sec\n",
            "[M::process] read 71414 sequences (10000053 bp)...\n",
            "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 34731, 0, 0)\n",
            "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (319, 339, 360)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (237, 442)\n",
            "[M::mem_pestat] mean and std.dev: (340.08, 28.37)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (196, 483)\n",
            "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
            "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
            "[M::mem_process_seqs] Processed 71454 reads in 21.667 CPU sec, 21.506 real sec\n",
            "[M::process] read 71436 sequences (10000101 bp)...\n",
            "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 34692, 0, 0)\n",
            "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (319, 339, 359)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (239, 439)\n",
            "[M::mem_pestat] mean and std.dev: (339.95, 28.44)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (199, 479)\n",
            "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
            "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
            "[M::mem_process_seqs] Processed 71414 reads in 20.645 CPU sec, 20.394 real sec\n",
            "[M::process] read 71428 sequences (10000197 bp)...\n",
            "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 34725, 0, 0)\n",
            "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (319, 339, 359)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (239, 439)\n",
            "[M::mem_pestat] mean and std.dev: (339.96, 28.22)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (199, 479)\n",
            "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
            "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
            "[M::mem_process_seqs] Processed 71436 reads in 21.420 CPU sec, 21.284 real sec\n",
            "[M::process] read 71430 sequences (10000071 bp)...\n",
            "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 34722, 0, 0)\n",
            "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (319, 339, 359)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (239, 439)\n",
            "[M::mem_pestat] mean and std.dev: (339.81, 28.06)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (199, 479)\n",
            "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
            "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
            "[M::mem_process_seqs] Processed 71428 reads in 21.704 CPU sec, 21.504 real sec\n",
            "[M::process] read 71426 sequences (10000066 bp)...\n",
            "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 34749, 0, 0)\n",
            "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (320, 339, 360)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (240, 440)\n",
            "[M::mem_pestat] mean and std.dev: (340.24, 28.34)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (200, 480)\n",
            "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
            "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
            "[M::mem_process_seqs] Processed 71430 reads in 20.767 CPU sec, 20.745 real sec\n",
            "[M::process] read 71438 sequences (10000081 bp)...\n",
            "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 34720, 0, 0)\n",
            "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (319, 339, 359)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (239, 439)\n",
            "[M::mem_pestat] mean and std.dev: (339.90, 28.19)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (199, 479)\n",
            "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
            "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
            "[M::mem_process_seqs] Processed 71426 reads in 21.531 CPU sec, 21.470 real sec\n",
            "[M::process] read 71426 sequences (10000055 bp)...\n",
            "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 34762, 0, 0)\n",
            "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (319, 339, 359)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (239, 439)\n",
            "[M::mem_pestat] mean and std.dev: (340.02, 28.30)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (199, 479)\n",
            "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
            "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
            "[M::mem_process_seqs] Processed 71438 reads in 20.999 CPU sec, 20.763 real sec\n",
            "[M::process] read 71430 sequences (10000131 bp)...\n",
            "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 34733, 0, 0)\n",
            "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (319, 339, 359)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (239, 439)\n",
            "[M::mem_pestat] mean and std.dev: (340.04, 28.28)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (199, 479)\n",
            "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
            "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
            "[M::mem_process_seqs] Processed 71426 reads in 21.367 CPU sec, 21.505 real sec\n",
            "[M::process] read 71426 sequences (10000195 bp)...\n",
            "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 34769, 0, 0)\n",
            "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (319, 339, 359)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (239, 439)\n",
            "[M::mem_pestat] mean and std.dev: (339.98, 28.28)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (199, 479)\n",
            "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
            "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
            "[M::mem_process_seqs] Processed 71430 reads in 21.578 CPU sec, 21.375 real sec\n",
            "[M::process] read 71430 sequences (10000036 bp)...\n",
            "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 34745, 0, 0)\n",
            "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (319, 339, 359)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (239, 439)\n",
            "[M::mem_pestat] mean and std.dev: (339.78, 28.20)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (199, 479)\n",
            "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
            "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
            "[M::mem_process_seqs] Processed 71426 reads in 20.698 CPU sec, 20.449 real sec\n",
            "[M::process] read 71400 sequences (10000135 bp)...\n",
            "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 34757, 0, 0)\n",
            "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (319, 339, 360)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (237, 442)\n",
            "[M::mem_pestat] mean and std.dev: (339.99, 28.48)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (196, 483)\n",
            "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
            "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
            "[M::mem_process_seqs] Processed 71430 reads in 21.254 CPU sec, 21.450 real sec\n",
            "[M::process] read 71464 sequences (10000254 bp)...\n",
            "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 34700, 0, 0)\n",
            "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (319, 339, 359)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (239, 439)\n",
            "[M::mem_pestat] mean and std.dev: (339.78, 28.33)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (199, 479)\n",
            "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
            "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
            "[M::mem_process_seqs] Processed 71400 reads in 21.583 CPU sec, 21.382 real sec\n",
            "[M::process] read 71396 sequences (10000284 bp)...\n",
            "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 34736, 0, 0)\n",
            "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (319, 339, 359)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (239, 439)\n",
            "[M::mem_pestat] mean and std.dev: (339.86, 28.49)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (199, 479)\n",
            "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
            "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
            "[M::mem_process_seqs] Processed 71464 reads in 20.769 CPU sec, 20.534 real sec\n",
            "[M::process] read 71454 sequences (10000005 bp)...\n",
            "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 34686, 0, 0)\n",
            "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (319, 339, 360)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (237, 442)\n",
            "[M::mem_pestat] mean and std.dev: (340.06, 28.42)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (196, 483)\n",
            "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
            "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
            "[M::mem_process_seqs] Processed 71396 reads in 21.567 CPU sec, 21.410 real sec\n",
            "[M::process] read 71436 sequences (10000230 bp)...\n",
            "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 34709, 0, 0)\n",
            "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (319, 339, 360)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (237, 442)\n",
            "[M::mem_pestat] mean and std.dev: (340.04, 28.36)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (196, 483)\n",
            "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
            "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
            "[M::mem_process_seqs] Processed 71454 reads in 21.596 CPU sec, 21.420 real sec\n",
            "[M::process] read 71418 sequences (10000140 bp)...\n",
            "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 34707, 0, 0)\n",
            "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (319, 339, 359)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (239, 439)\n",
            "[M::mem_pestat] mean and std.dev: (339.90, 28.14)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (199, 479)\n",
            "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
            "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
            "[M::mem_process_seqs] Processed 71436 reads in 20.371 CPU sec, 20.099 real sec\n",
            "[M::process] read 71400 sequences (10000124 bp)...\n",
            "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 34742, 0, 0)\n",
            "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (319, 339, 359)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (239, 439)\n",
            "[M::mem_pestat] mean and std.dev: (339.69, 28.19)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (199, 479)\n",
            "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
            "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
            "[M::mem_process_seqs] Processed 71418 reads in 21.929 CPU sec, 21.795 real sec\n",
            "[M::process] read 71420 sequences (10000002 bp)...\n",
            "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 34711, 0, 0)\n",
            "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (319, 339, 359)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (239, 439)\n",
            "[M::mem_pestat] mean and std.dev: (339.78, 28.26)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (199, 479)\n",
            "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
            "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
            "[M::mem_process_seqs] Processed 71400 reads in 21.736 CPU sec, 21.615 real sec\n",
            "[M::process] read 71402 sequences (10000259 bp)...\n",
            "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 34774, 0, 0)\n",
            "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (319, 339, 359)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (239, 439)\n",
            "[M::mem_pestat] mean and std.dev: (339.87, 28.05)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (199, 479)\n",
            "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
            "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
            "[M::mem_process_seqs] Processed 71420 reads in 20.777 CPU sec, 20.517 real sec\n",
            "[M::process] read 71410 sequences (10000193 bp)...\n",
            "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 34754, 0, 0)\n",
            "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (320, 339, 359)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (242, 437)\n",
            "[M::mem_pestat] mean and std.dev: (340.13, 28.19)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (203, 476)\n",
            "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
            "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
            "[M::mem_process_seqs] Processed 71402 reads in 21.395 CPU sec, 21.225 real sec\n",
            "[M::process] read 71460 sequences (10000088 bp)...\n",
            "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 34740, 0, 0)\n",
            "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (319, 339, 359)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (239, 439)\n",
            "[M::mem_pestat] mean and std.dev: (339.85, 28.25)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (199, 479)\n",
            "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
            "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
            "[M::mem_process_seqs] Processed 71410 reads in 21.653 CPU sec, 21.415 real sec\n",
            "[M::process] read 71418 sequences (10000274 bp)...\n",
            "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 34733, 0, 0)\n",
            "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (319, 339, 359)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (239, 439)\n",
            "[M::mem_pestat] mean and std.dev: (339.99, 28.42)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (199, 479)\n",
            "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
            "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
            "[M::mem_process_seqs] Processed 71460 reads in 20.710 CPU sec, 20.562 real sec\n",
            "[M::process] read 71464 sequences (10000278 bp)...\n",
            "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 34708, 0, 0)\n",
            "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (319, 339, 359)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (239, 439)\n",
            "[M::mem_pestat] mean and std.dev: (339.87, 28.33)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (199, 479)\n",
            "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
            "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
            "[M::mem_process_seqs] Processed 71418 reads in 24.444 CPU sec, 26.033 real sec\n",
            "[M::process] read 71416 sequences (10000161 bp)...\n",
            "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 34779, 0, 0)\n",
            "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (319, 339, 359)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (239, 439)\n",
            "[M::mem_pestat] mean and std.dev: (339.95, 28.44)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (199, 479)\n",
            "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
            "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
            "[M::mem_process_seqs] Processed 71464 reads in 21.287 CPU sec, 21.043 real sec\n",
            "[M::process] read 71476 sequences (10000115 bp)...\n",
            "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 34785, 0, 0)\n",
            "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (319, 339, 359)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (239, 439)\n",
            "[M::mem_pestat] mean and std.dev: (339.85, 28.23)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (199, 479)\n",
            "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
            "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
            "[M::mem_process_seqs] Processed 71416 reads in 20.210 CPU sec, 19.923 real sec\n",
            "[M::process] read 71434 sequences (10000196 bp)...\n",
            "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 34720, 0, 0)\n",
            "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (319, 339, 359)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (239, 439)\n",
            "[M::mem_pestat] mean and std.dev: (339.62, 28.28)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (199, 479)\n",
            "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
            "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
            "[M::mem_process_seqs] Processed 71476 reads in 21.444 CPU sec, 21.341 real sec\n",
            "[M::process] read 71398 sequences (10000300 bp)...\n",
            "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 34695, 0, 0)\n",
            "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (319, 339, 359)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (239, 439)\n",
            "[M::mem_pestat] mean and std.dev: (339.61, 28.17)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (199, 479)\n",
            "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
            "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
            "[M::mem_process_seqs] Processed 71434 reads in 21.241 CPU sec, 21.054 real sec\n",
            "[M::process] read 71420 sequences (10000253 bp)...\n",
            "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 34694, 0, 0)\n",
            "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (319, 339, 359)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (239, 439)\n",
            "[M::mem_pestat] mean and std.dev: (340.09, 28.26)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (199, 479)\n",
            "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
            "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
            "[M::mem_process_seqs] Processed 71398 reads in 20.589 CPU sec, 20.306 real sec\n",
            "[M::process] read 71390 sequences (10000051 bp)...\n",
            "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 34703, 0, 0)\n",
            "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (319, 339, 359)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (239, 439)\n",
            "[M::mem_pestat] mean and std.dev: (339.74, 28.35)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (199, 479)\n",
            "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
            "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
            "[M::mem_process_seqs] Processed 71420 reads in 21.356 CPU sec, 21.183 real sec\n",
            "[M::process] read 71422 sequences (10000184 bp)...\n",
            "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 34739, 0, 0)\n",
            "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (319, 339, 359)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (239, 439)\n",
            "[M::mem_pestat] mean and std.dev: (340.15, 28.20)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (199, 479)\n",
            "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
            "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
            "[M::mem_process_seqs] Processed 71390 reads in 21.521 CPU sec, 21.318 real sec\n",
            "[M::process] read 71412 sequences (10000289 bp)...\n",
            "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 34768, 0, 0)\n",
            "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (319, 339, 359)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (239, 439)\n",
            "[M::mem_pestat] mean and std.dev: (339.96, 28.29)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (199, 479)\n",
            "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
            "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
            "[M::mem_process_seqs] Processed 71422 reads in 20.252 CPU sec, 19.993 real sec\n",
            "[M::process] read 71402 sequences (10000196 bp)...\n",
            "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 34738, 0, 0)\n",
            "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (319, 339, 359)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (239, 439)\n",
            "[M::mem_pestat] mean and std.dev: (339.75, 28.10)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (199, 479)\n",
            "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
            "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
            "[M::mem_process_seqs] Processed 71412 reads in 21.481 CPU sec, 21.244 real sec\n",
            "[M::process] read 71442 sequences (10000137 bp)...\n",
            "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 34756, 0, 0)\n",
            "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (319, 339, 359)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (239, 439)\n",
            "[M::mem_pestat] mean and std.dev: (339.67, 28.23)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (199, 479)\n",
            "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
            "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
            "[M::mem_process_seqs] Processed 71402 reads in 21.085 CPU sec, 20.886 real sec\n",
            "[M::process] read 71444 sequences (10000135 bp)...\n",
            "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 34722, 0, 0)\n",
            "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (319, 339, 359)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (239, 439)\n",
            "[M::mem_pestat] mean and std.dev: (339.89, 28.45)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (199, 479)\n",
            "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
            "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
            "[M::mem_process_seqs] Processed 71442 reads in 21.074 CPU sec, 21.152 real sec\n",
            "[M::process] read 71448 sequences (10000005 bp)...\n",
            "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 34781, 0, 0)\n",
            "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (319, 339, 359)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (239, 439)\n",
            "[M::mem_pestat] mean and std.dev: (339.88, 28.43)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (199, 479)\n",
            "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
            "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
            "[M::mem_process_seqs] Processed 71444 reads in 21.542 CPU sec, 21.336 real sec\n",
            "[M::process] read 71444 sequences (10000284 bp)...\n",
            "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 34725, 0, 0)\n",
            "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (319, 339, 359)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (239, 439)\n",
            "[M::mem_pestat] mean and std.dev: (339.75, 28.19)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (199, 479)\n",
            "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
            "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
            "[M::mem_process_seqs] Processed 71448 reads in 20.720 CPU sec, 20.465 real sec\n",
            "[M::process] read 71440 sequences (10000195 bp)...\n",
            "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 34716, 0, 0)\n",
            "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (319, 339, 359)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (239, 439)\n",
            "[M::mem_pestat] mean and std.dev: (339.71, 28.31)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (199, 479)\n",
            "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
            "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
            "[M::mem_process_seqs] Processed 71444 reads in 21.309 CPU sec, 21.467 real sec\n",
            "[M::process] read 71416 sequences (10000256 bp)...\n",
            "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 34749, 0, 0)\n",
            "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (319, 339, 359)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (239, 439)\n",
            "[M::mem_pestat] mean and std.dev: (339.68, 28.26)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (199, 479)\n",
            "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
            "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
            "[M::mem_process_seqs] Processed 71440 reads in 21.347 CPU sec, 21.158 real sec\n",
            "[M::process] read 71444 sequences (10000220 bp)...\n",
            "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 34741, 0, 0)\n",
            "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (319, 339, 359)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (239, 439)\n",
            "[M::mem_pestat] mean and std.dev: (339.86, 28.25)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (199, 479)\n",
            "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
            "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
            "[M::mem_process_seqs] Processed 71416 reads in 20.602 CPU sec, 20.356 real sec\n",
            "[M::process] read 71456 sequences (10000138 bp)...\n",
            "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 34746, 0, 0)\n",
            "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (319, 339, 359)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (239, 439)\n",
            "[M::mem_pestat] mean and std.dev: (339.97, 28.28)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (199, 479)\n",
            "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
            "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
            "[M::mem_process_seqs] Processed 71444 reads in 21.432 CPU sec, 21.230 real sec\n",
            "[M::process] read 71402 sequences (9994959 bp)...\n",
            "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 34739, 0, 0)\n",
            "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (319, 339, 359)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (239, 439)\n",
            "[M::mem_pestat] mean and std.dev: (339.87, 28.32)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (199, 479)\n",
            "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
            "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
            "[M::mem_process_seqs] Processed 71456 reads in 21.332 CPU sec, 21.164 real sec\n",
            "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 34786, 0, 0)\n",
            "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (319, 339, 359)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (239, 439)\n",
            "[M::mem_pestat] mean and std.dev: (339.77, 28.28)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (199, 479)\n",
            "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
            "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
            "[M::mem_process_seqs] Processed 71402 reads in 19.941 CPU sec, 19.997 real sec\n",
            "[main] Version: 0.7.18-r1243-dirty\n",
            "[main] CMD: bwa mem -R @RG\\tID:sample1\\tSM:sample1\\tPL:ILLUMINA\\tLB:lib1\\tPU:unit1 reference/Chr12-reference.fasta cutadapt/caso2-sindrome-metabolica-trimmed_R1.fastq.gz cutadapt/caso2-sindrome-metabolica-trimmed_R2.fastq.gz\n",
            "[main] Real time: 1797.725 sec; CPU: 1806.941 sec\n",
            "[bam_sort_core] merging from 2 files and 1 in-memory blocks...\n"
          ]
        }
      ],
      "source": [
        "%%bash\n",
        "# Proceder com o alinhamento contra o genoma humano\n",
        "bwa mem reference/Chr12-reference.fasta \\\n",
        "  -R '@RG\\tID:sample1\\tSM:sample1\\tPL:ILLUMINA\\tLB:lib1\\tPU:unit1' \\\n",
        "  cutadapt/caso2-sindrome-metabolica-trimmed_R1.fastq.gz \\\n",
        "  cutadapt/caso2-sindrome-metabolica-trimmed_R2.fastq.gz \\\n",
        "  | samtools sort -o bwa/caso2-sindrome-metabolica-mapped-host.bam"
      ]
    },
    {
      "cell_type": "code",
      "source": [
        "%%bash\n",
        "# Indexar o bam para gerar o bai e visualizar no IGV\n",
        "samtools index bwa/caso2-sindrome-metabolica-mapped-host.bam"
      ],
      "metadata": {
        "id": "rjSUyNMaKUQO"
      },
      "execution_count": null,
      "outputs": []
    },
    {
      "cell_type": "markdown",
      "source": [
        "**Checkpoint 2:**  \n",
        "Observando os arquivos .BAM e .BAI no IGV. Referenciando com o gene do Chr12 no gene PAH.\n",
        "\n",
        "Olhando no IGV, foi possível analizar que existem várias trocas de base e algumas mutações, sendo a maioria delas em íntrons e duas em éxons"
      ],
      "metadata": {
        "id": "mnEpEJDuKcXD"
      }
    },
    {
      "cell_type": "markdown",
      "source": [
        "# Etapa 3 - Chamada de Variantes"
      ],
      "metadata": {
        "id": "mq01gI8l8Inn"
      }
    },
    {
      "cell_type": "code",
      "source": [
        "%%bash\n",
        "# Indexar a referência e BAM para chamada de variantes com GATK\n",
        "samtools dict reference/Chr12-reference.fasta > reference/Chr12-reference.dict\n",
        "samtools index bwa/caso2-sindrome-metabolica-mapped-host.bam"
      ],
      "metadata": {
        "id": "s5Qq7DWx-uUO"
      },
      "execution_count": null,
      "outputs": []
    },
    {
      "cell_type": "code",
      "source": [
        "%%bash\n",
        "# Chamada de variantes\n",
        "gatk HaplotypeCaller \\\n",
        "  -R reference/Chr12-reference.fasta \\\n",
        "  -I bwa/caso2-sindrome-metabolica-mapped-host.bam \\\n",
        "  -O gatk/caso2-sindrome-metabolica-host-variants.vcf"
      ],
      "metadata": {
        "id": "6R5-xndf8M9S",
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "collapsed": true,
        "outputId": "d5fc761f-3753-48c9-ad42-a91e245f1279"
      },
      "execution_count": null,
      "outputs": [
        {
          "output_type": "stream",
          "name": "stderr",
          "text": [
            "Using GATK jar /usr/local/share/gatk4-4.6.2.0-0/gatk-package-4.6.2.0-local.jar\n",
            "Running:\n",
            "    java -Dsamjdk.use_async_io_read_samtools=false -Dsamjdk.use_async_io_write_samtools=true -Dsamjdk.use_async_io_write_tribble=false -Dsamjdk.compression_level=2 -jar /usr/local/share/gatk4-4.6.2.0-0/gatk-package-4.6.2.0-local.jar HaplotypeCaller -R reference/Chr12-reference.fasta -I bwa/caso2-sindrome-metabolica-mapped-host.bam -O gatk/caso2-sindrome-metabolica-host-variants.vcf\n",
            "01:45:26.645 INFO  NativeLibraryLoader - Loading libgkl_compression.so from jar:file:/usr/local/share/gatk4-4.6.2.0-0/gatk-package-4.6.2.0-local.jar!/com/intel/gkl/native/libgkl_compression.so\n",
            "01:45:26.902 INFO  HaplotypeCaller - ------------------------------------------------------------\n",
            "01:45:26.909 INFO  HaplotypeCaller - The Genome Analysis Toolkit (GATK) v4.6.2.0\n",
            "01:45:26.909 INFO  HaplotypeCaller - For support and documentation go to https://software.broadinstitute.org/gatk/\n",
            "01:45:26.910 INFO  HaplotypeCaller - Executing as root@d4aa53e725f4 on Linux v6.6.105+ amd64\n",
            "01:45:26.910 INFO  HaplotypeCaller - Java runtime: OpenJDK 64-Bit Server VM v17.0.11-internal+0-adhoc..src\n",
            "01:45:26.911 INFO  HaplotypeCaller - Start Date/Time: October 31, 2025 at 1:45:26 AM UTC\n",
            "01:45:26.911 INFO  HaplotypeCaller - ------------------------------------------------------------\n",
            "01:45:26.911 INFO  HaplotypeCaller - ------------------------------------------------------------\n",
            "01:45:26.913 INFO  HaplotypeCaller - HTSJDK Version: 4.2.0\n",
            "01:45:26.914 INFO  HaplotypeCaller - Picard Version: 3.4.0\n",
            "01:45:26.914 INFO  HaplotypeCaller - Built for Spark Version: 3.5.0\n",
            "01:45:26.918 INFO  HaplotypeCaller - HTSJDK Defaults.COMPRESSION_LEVEL : 2\n",
            "01:45:26.919 INFO  HaplotypeCaller - HTSJDK Defaults.USE_ASYNC_IO_READ_FOR_SAMTOOLS : false\n",
            "01:45:26.919 INFO  HaplotypeCaller - HTSJDK Defaults.USE_ASYNC_IO_WRITE_FOR_SAMTOOLS : true\n",
            "01:45:26.920 INFO  HaplotypeCaller - HTSJDK Defaults.USE_ASYNC_IO_WRITE_FOR_TRIBBLE : false\n",
            "01:45:26.920 INFO  HaplotypeCaller - Deflater: IntelDeflater\n",
            "01:45:26.921 INFO  HaplotypeCaller - Inflater: IntelInflater\n",
            "01:45:26.921 INFO  HaplotypeCaller - GCS max retries/reopens: 20\n",
            "01:45:26.921 INFO  HaplotypeCaller - Requester pays: disabled\n",
            "01:45:26.922 INFO  HaplotypeCaller - Initializing engine\n",
            "01:45:27.206 INFO  HaplotypeCaller - Done initializing engine\n",
            "01:45:27.224 INFO  NativeLibraryLoader - Loading libgkl_utils.so from jar:file:/usr/local/share/gatk4-4.6.2.0-0/gatk-package-4.6.2.0-local.jar!/com/intel/gkl/native/libgkl_utils.so\n",
            "01:45:27.235 INFO  NativeLibraryLoader - Loading libgkl_smithwaterman.so from jar:file:/usr/local/share/gatk4-4.6.2.0-0/gatk-package-4.6.2.0-local.jar!/com/intel/gkl/native/libgkl_smithwaterman.so\n",
            "01:45:27.238 INFO  SmithWatermanAligner - Using AVX accelerated SmithWaterman implementation\n",
            "01:45:27.243 INFO  HaplotypeCallerEngine - Disabling physical phasing, which is supported only for reference-model confidence output\n",
            "01:45:27.261 INFO  NativeLibraryLoader - Loading libgkl_pairhmm_omp.so from jar:file:/usr/local/share/gatk4-4.6.2.0-0/gatk-package-4.6.2.0-local.jar!/com/intel/gkl/native/libgkl_pairhmm_omp.so\n",
            "01:45:27.284 INFO  IntelPairHmm - Flush-to-zero (FTZ) is enabled when running PairHMM\n",
            "01:45:27.285 INFO  IntelPairHmm - Available threads: 2\n",
            "01:45:27.285 INFO  IntelPairHmm - Requested threads: 4\n",
            "01:45:27.285 WARN  IntelPairHmm - Using 2 available threads, but 4 were requested\n",
            "01:45:27.286 INFO  PairHMM - Using the OpenMP multi-threaded AVX-accelerated native PairHMM implementation\n",
            "01:45:27.363 INFO  ProgressMeter - Starting traversal\n",
            "01:45:27.364 INFO  ProgressMeter -        Current Locus  Elapsed Minutes     Regions Processed   Regions/Minute\n",
            "01:45:37.369 INFO  ProgressMeter - NC_000012.12:36917288              0.2                123060         738360.0\n",
            "01:45:47.372 INFO  ProgressMeter - NC_000012.12:82880034              0.3                276270         828602.8\n",
            "01:45:52.692 WARN  InbreedingCoeff - InbreedingCoeff will not be calculated at position NC_000012.12:83100144 and possibly subsequent; at least 10 samples must have called genotypes\n",
            "01:45:57.380 INFO  ProgressMeter - NC_000012.12:83496791              0.5                278380         556481.8\n",
            "01:46:07.387 INFO  ProgressMeter - NC_000012.12:84350828              0.7                281330         421752.5\n",
            "01:46:17.389 INFO  ProgressMeter - NC_000012.12:85594613              0.8                285590         342543.6\n",
            "01:46:27.405 INFO  ProgressMeter - NC_000012.12:87383955              1.0                291730         291530.8\n",
            "01:46:37.424 INFO  ProgressMeter - NC_000012.12:89026269              1.2                297360         254665.4\n",
            "01:46:47.434 INFO  ProgressMeter - NC_000012.12:90600026              1.3                302800         226904.3\n",
            "01:46:57.445 INFO  ProgressMeter - NC_000012.12:92085163              1.5                307910         205091.0\n",
            "01:47:07.446 INFO  ProgressMeter - NC_000012.12:93810455              1.7                313870         188169.6\n",
            "01:47:17.445 INFO  ProgressMeter - NC_000012.12:95477819              1.8                319610         174204.4\n",
            "01:47:27.448 INFO  ProgressMeter - NC_000012.12:96865978              2.0                324420         162096.5\n",
            "01:47:37.457 INFO  ProgressMeter - NC_000012.12:98486548              2.2                329970         152185.0\n",
            "01:47:47.471 INFO  ProgressMeter - NC_000012.12:100336277              2.3                336330         144031.3\n",
            "01:47:57.473 INFO  ProgressMeter - NC_000012.12:102118258              2.5                342460         136885.4\n",
            "01:48:07.486 INFO  ProgressMeter - NC_000012.12:103803477              2.7                348230         130486.8\n",
            "01:48:17.496 INFO  ProgressMeter - NC_000012.12:105448526              2.8                353880         124801.9\n",
            "01:48:27.502 INFO  ProgressMeter - NC_000012.12:107298148              3.0                360270         119998.0\n",
            "01:48:37.508 INFO  ProgressMeter - NC_000012.12:109154879              3.2                366620         115687.1\n",
            "01:48:47.510 INFO  ProgressMeter - NC_000012.12:110839929              3.3                372430         111647.5\n",
            "01:48:57.512 INFO  ProgressMeter - NC_000012.12:112522952              3.5                378220         107986.8\n",
            "01:49:07.521 INFO  ProgressMeter - NC_000012.12:114355681              3.7                384530         104797.0\n",
            "01:49:17.532 INFO  ProgressMeter - NC_000012.12:116229386              3.8                390950         101913.0\n",
            "01:49:27.535 INFO  ProgressMeter - NC_000012.12:117966791              4.0                396940          99164.3\n",
            "01:49:37.536 INFO  ProgressMeter - NC_000012.12:119622250              4.2                402630          96564.8\n",
            "01:49:47.562 INFO  ProgressMeter - NC_000012.12:121412495              4.3                408780          94262.1\n",
            "01:49:57.562 INFO  ProgressMeter - NC_000012.12:124895680              4.5                420570          93391.5\n",
            "01:50:00.065 INFO  HaplotypeCaller - 7165 read(s) filtered by: MappingQualityReadFilter \n",
            "0 read(s) filtered by: MappingQualityAvailableReadFilter \n",
            "0 read(s) filtered by: MappedReadFilter \n",
            "0 read(s) filtered by: NotSecondaryAlignmentReadFilter \n",
            "0 read(s) filtered by: NotDuplicateReadFilter \n",
            "0 read(s) filtered by: PassesVendorQualityCheckReadFilter \n",
            "0 read(s) filtered by: NonZeroReferenceLengthAlignmentReadFilter \n",
            "0 read(s) filtered by: GoodCigarReadFilter \n",
            "0 read(s) filtered by: WellformedReadFilter \n",
            "7165 total reads filtered out of 5999996 reads processed\n",
            "01:50:00.066 INFO  ProgressMeter - NC_000012.12:133274680              4.5                448502          98679.6\n",
            "01:50:00.066 INFO  ProgressMeter - Traversal complete. Processed 448502 total regions in 4.5 minutes.\n",
            "01:50:00.185 INFO  VectorLoglessPairHMM - Time spent in setup for JNI call : 0.048506519000000005\n",
            "01:50:00.185 INFO  PairHMM - Total compute time in PairHMM computeLogLikelihoods() : 3.2638714330000003\n",
            "01:50:00.186 INFO  SmithWatermanAligner - Total compute time in native Smith-Waterman : 8.00 sec\n",
            "01:50:00.187 INFO  HaplotypeCaller - Shutting down engine\n",
            "[October 31, 2025 at 1:50:00 AM UTC] org.broadinstitute.hellbender.tools.walkers.haplotypecaller.HaplotypeCaller done. Elapsed time: 4.56 minutes.\n",
            "Runtime.totalMemory()=641728512\n"
          ]
        }
      ]
    },
    {
      "cell_type": "markdown",
      "source": [
        "# Etapa 4 - Anotação de Variantes"
      ],
      "metadata": {
        "id": "7y2nvBXX8mhL"
      }
    },
    {
      "cell_type": "code",
      "source": [
        "%%bash\n",
        "# script oficial baixa cache do genoma humano\n",
        "vep_install -a cf -s homo_sapiens -y GRCh38 --NO_UPDATE"
      ],
      "metadata": {
        "id": "bfXldub7TADL",
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "outputId": "4c278078-23dd-49a1-d949-c414cac15fef",
        "collapsed": true
      },
      "execution_count": null,
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            " - getting list of available cache files\n",
            " - downloading https://ftp.ensembl.org/pub/release-115/variation/indexed_vep_cache/homo_sapiens_vep_115_GRCh38.tar.gz\n",
            " - unpacking homo_sapiens_vep_115_GRCh38.tar.gz\n",
            " - converting cache, this may take some time but will allow VEP to look up variants and frequency data much faster\n",
            " - use CTRL-C to cancel if you do not wish to convert this cache now (you may run convert_cache.pl later)\n",
            "2025-10-31 02:33:53 - Processing homo_sapiens\n",
            "2025-10-31 02:33:53 - Processing version 115_GRCh38\n",
            "2025-10-31 02:33:53 - No unprocessed types remaining, skipping\n",
            "2025-10-31 02:33:53 - All done!\n",
            " - downloading Homo_sapiens.GRCh38.dna.toplevel.fa.gz\n",
            " - downloading Homo_sapiens.GRCh38.dna.toplevel.fa.gz.fai\n",
            " - downloading Homo_sapiens.GRCh38.dna.toplevel.fa.gz.gzi\n",
            "\n",
            "The FASTA file should be automatically detected by the VEP when using --cache or --offline.\n",
            "If it is not, use \"--fasta /root/.vep/homo_sapiens/115_GRCh38/Homo_sapiens.GRCh38.dna.toplevel.fa.gz\"\n",
            "\n",
            "\n",
            "All done\n"
          ]
        },
        {
          "output_type": "stream",
          "name": "stderr",
          "text": [
            "Unable to close datastream at /usr/local/bin/vep_install line 1385.\n"
          ]
        }
      ]
    },
    {
      "cell_type": "code",
      "source": [
        "%%bash\n",
        "# Index VCF files para anotação\n",
        "bgzip gatk/caso2-sindrome-metabolica-host-variants.vcf\n",
        "tabix -p vcf gatk/caso2-sindrome-metabolica-host-variants.vcf.gz\n"
      ],
      "metadata": {
        "id": "bvfu-FJiUohO"
      },
      "execution_count": null,
      "outputs": []
    },
    {
      "cell_type": "code",
      "source": [
        "%%bash\n",
        "# Anotação de variantes a nível de proteína\n",
        "vep -i gatk/caso2-sindrome-metabolica-host-variants.vcf.gz \\\n",
        "    --output_file vep/caso2-sindrome-metabolica-host-variants.vep.tsv \\\n",
        "    --tab --force_overwrite \\\n",
        "    --offline --cache --dir_cache ~/.vep \\\n",
        "    --assembly GRCh38 \\\n",
        "    --symbol --hgvs --canonical --af_1kg --max_af \\\n",
        "    --everything --pick\\\n",
        "    --stats_file vep/caso2-sindrome-metabolica-host-variants.vep.html\n"
      ],
      "metadata": {
        "id": "ZoUmcPRPxQ0Y"
      },
      "execution_count": null,
      "outputs": []
    },
    {
      "cell_type": "markdown",
      "source": [
        "# Etapa 5 - Predição de efeito"
      ],
      "metadata": {
        "id": "_Ymki9N2b29B"
      }
    },
    {
      "cell_type": "markdown",
      "source": [
        "**Checkpoint**:   \n",
        "- Use as informações de anotação presentes no arquivo tsv para inferir se as variantes são benignas ou de efeito deletério.\n",
        "\n",
        "\n",
        "```\n",
        "Resp:\n",
        "```\n",
        "\n",
        "\n",
        "- Encontre a frequência desta mutação em populações americanas e europeias:  \n",
        "\n",
        "```\n",
        "Resp:\n",
        "```\n",
        "\n"
      ],
      "metadata": {
        "id": "l6txySPXolx7"
      }
    },
    {
      "cell_type": "markdown",
      "source": [
        "Que outras informações você poderia obter para melhorar a sua conclusão sobre a patogenicidade das variantes?  \n",
        "\n",
        "\n",
        "```\n",
        "Resp:\n",
        "```\n",
        "\n"
      ],
      "metadata": {
        "id": "GVEaoL3zpTmK"
      }
    },
    {
      "cell_type": "markdown",
      "source": [
        "Caso tenha curiosidade, use o Google AlphaFold para predizer a estrutura 3D da proteína alterada: [Colabs](https://colab.research.google.com/drive/1lA0GqBdptSsBylRidY0QJPjcKIy3R45c?usp=sharing)"
      ],
      "metadata": {
        "id": "B3Q81YhSFaBR"
      }
    }
  ],
  "metadata": {
    "colab": {
      "provenance": []
    },
    "kernelspec": {
      "display_name": "Python 3",
      "name": "python3"
    },
    "language_info": {
      "name": "python"
    }
  },
  "nbformat": 4,
  "nbformat_minor": 0
}